--- config/gnu-flags.orig	2012-04-13 17:12:00.000000000 +0200
+++ config/gnu-flags	2012-04-17 18:07:25.104448452 +0200
@@ -325,6 +325,9 @@
     # (Strictly speaking this isn't really a "warning" flag, so it's added to
     #   the debugging flags)
     #DEBUG_CFLAGS="$DEBUG_CFLAGS -Wstack-protector -fstack-protector-all"
+
+    # Enable some additional optimization settings
+    PROD_CFLAGS="$PROD_CFLAGS -floop-interchange -floop-strip-mine -floop-block -fgraphite-identity -fuse-linker-plugin -flto -mtune=corei7"
     ;;
 
   gcc-4.5*)
