diff -ruN hdf5-1.8.13.orig/config/gnu-flags hdf5-1.8.13/config/gnu-flags
--- config/gnu-flags	2014-05-06 04:13:21.000000000 +0200
+++ config/gnu-flags	2014-06-06 15:36:09.318183153 +0200
@@ -260,6 +260,9 @@
     # (Strictly speaking this isn't really a "warning" flag, so it's added to
     #   the debugging flags)
     #DEBUG_CFLAGS="$DEBUG_CFLAGS -Wstack-protector -fstack-protector-all"
+
+    # Enable some additional optimization settings
+    PROD_CFLAGS="$PROD_CFLAGS -floop-interchange -floop-strip-mine -floop-block -fgraphite-identity -fuse-linker-plugin -flto -mtune=corei7"
     ;;
 
   gcc-4.7*)
@@ -331,6 +334,9 @@
     # (Strictly speaking this isn't really a "warning" flag, so it's added to
     #   the debugging flags)
     #DEBUG_CFLAGS="$DEBUG_CFLAGS -Wstack-protector -fstack-protector-all"
+
+    # Enable some additional optimization settings
+    PROD_CFLAGS="$PROD_CFLAGS -floop-interchange -floop-strip-mine -floop-block -fgraphite-identity -fuse-linker-plugin -flto -mtune=corei7"
     ;;
 
   gcc-4.6*)
@@ -395,6 +401,9 @@
     # (Strictly speaking this isn't really a "warning" flag, so it's added to
     #   the debugging flags)
     #DEBUG_CFLAGS="$DEBUG_CFLAGS -Wstack-protector -fstack-protector-all"
+
+    # Enable some additional optimization settings
+    PROD_CFLAGS="$PROD_CFLAGS -floop-interchange -floop-strip-mine -floop-block -fgraphite-identity -fuse-linker-plugin -flto -mtune=corei7"
     ;;
 
   gcc-4.5*)
