--- a/config/gnu-flags	2015-10-24 07:14:05.000000000 +0200
+++ b/config/gnu-flags	2016-02-20 21:37:33.000000000 +0100
@@ -99,10 +99,12 @@
     # Production
     case "$cc_vendor-$cc_version" in
       gcc-[34].*)
-        PROD_CFLAGS="-O3"
+        # Enable some additional optimization settings
+        PROD_CFLAGS="-O3 -floop-interchange -floop-strip-mine -floop-block -fgraphite-identity -fuse-linker-plugin -flto -mtune=corei7"
         ;;
       gcc-5.*)
-        PROD_CFLAGS="-O3 -fstdarg-opt"
+        # Enable some additional optimization settings
+        PROD_CFLAGS="-O3 -fstdarg-opt -floop-interchange -floop-strip-mine -floop-block -fgraphite-identity -fuse-linker-plugin -flto -mtune=corei7"
         ;;
       *)
         PROD_CFLAGS="-O -finline-functions"
