--- config/gnu-flags.orig	2011-04-23 18:22:36.000000000 +0200
+++ config/gnu-flags	2011-04-23 18:26:42.000000000 +0200
@@ -245,6 +245,9 @@
     # (Strictly speaking this isn't really a "warning" flag, so it's added to
     #   the debugging flags)
     #DEBUG_CFLAGS="$DEBUG_CFLAGS -Wstack-protector -fstack-protector-all"
+
+    # Enable some additional optimization settings
+    PROD_CFLAGS="$PROD_CFLAGS -floop-interchange -floop-strip-mine -floop-block -fgraphite-identity -fuse-linker-plugin -flto"
     ;;
 
   gcc-4.5*)
