Index: test/CMakeLists.txt
===================================================================
--- test/CMakeLists.txt	(revision 23770)
+++ test/CMakeLists.txt	(revision 23771)
@@ -27,6 +27,7 @@
 )
 
 ADD_LIBRARY (${HDF5_TEST_LIB_TARGET} ${LIB_TYPE} ${TEST_LIB_SRCS} ${TEST_LIB_HEADERS})
+TARGET_C_PROPERTIES (${HDF5_TEST_LIB_TARGET} " " " ")
 IF (MSVC)
   TARGET_LINK_LIBRARIES (${HDF5_TEST_LIB_TARGET} "ws2_32.lib")
 ENDIF (MSVC)
@@ -64,6 +65,7 @@
     INCLUDE_DIRECTORIES (${HDF5_SRC_DIR})
 
     ADD_LIBRARY (${HDF5_TEST_PLUGIN_LIB_TARGET} ${LIB_TYPE} ${HDF5_TEST_SOURCE_DIR}/${test_lib}.c)
+   TARGET_C_PROPERTIES (${HDF5_TEST_PLUGIN_LIB_TARGET} " " " ")
     TARGET_LINK_LIBRARIES (${HDF5_TEST_PLUGIN_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
     H5_SET_LIB_OPTIONS (
         ${HDF5_TEST_PLUGIN_LIB_TARGET} ${HDF5_TEST_PLUGIN_LIB_NAME}
@@ -94,6 +96,7 @@
     INCLUDE_DIRECTORIES (${HDF5_SRC_DIR})
 
     ADD_LIBRARY (${HDF5_TEST_PLUGIN_LIB_TARGET} ${LIB_TYPE} ${HDF5_TEST_SOURCE_DIR}/${test_lib}.c)
+    TARGET_C_PROPERTIES (${HDF5_TEST_PLUGIN_LIB_TARGET} " " " ")
     TARGET_LINK_LIBRARIES (${HDF5_TEST_PLUGIN_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
     H5_SET_LIB_OPTIONS (
         ${HDF5_TEST_PLUGIN_LIB_TARGET} ${HDF5_TEST_PLUGIN_LIB_NAME}
@@ -237,6 +240,7 @@
 #-- Adding test for testhdf5
 ADD_EXECUTABLE (testhdf5 ${testhdf5_SRCS})
 TARGET_NAMING (testhdf5 ${LIB_TYPE})
+TARGET_C_PROPERTIES (testhdf5 " " " ")
 TARGET_LINK_LIBRARIES (testhdf5 ${HDF5_TEST_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (testhdf5 PROPERTIES FOLDER test)
 
@@ -288,6 +292,7 @@
 MACRO (ADD_H5_TEST file)
   ADD_EXECUTABLE (${file} ${HDF5_TEST_SOURCE_DIR}/${file}.c)
   TARGET_NAMING (${file} ${LIB_TYPE})
+  TARGET_C_PROPERTIES (${file} " " " ")
   TARGET_LINK_LIBRARIES (${file} ${HDF5_TEST_LIB_TARGET} ${HDF5_LIB_TARGET})
   SET_TARGET_PROPERTIES (${file} PROPERTIES FOLDER test)
 
@@ -428,6 +433,7 @@
 #-- Adding test for cache
 ADD_EXECUTABLE (cache ${HDF5_TEST_SOURCE_DIR}/cache.c ${HDF5_TEST_SOURCE_DIR}/cache_common.c)
 TARGET_NAMING (cache ${LIB_TYPE})
+TARGET_C_PROPERTIES (cache " " " ")
 TARGET_LINK_LIBRARIES (cache ${HDF5_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
 SET_TARGET_PROPERTIES (cache PROPERTIES FOLDER test)
 ADD_TEST (
@@ -442,6 +448,7 @@
 #-- Adding test for cache_api
 ADD_EXECUTABLE (cache_api ${HDF5_TEST_SOURCE_DIR}/cache_api.c ${HDF5_TEST_SOURCE_DIR}/cache_common.c)
 TARGET_NAMING (cache_api ${LIB_TYPE})
+TARGET_C_PROPERTIES (cache_api " " " ")
 TARGET_LINK_LIBRARIES (cache_api ${HDF5_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
 SET_TARGET_PROPERTIES (cache_api PROPERTIES FOLDER test)
 
@@ -463,6 +470,7 @@
     ${HDF5_TEST_SOURCE_DIR}/ttsafe_acreate.c
 )
 TARGET_NAMING (ttsafe ${LIB_TYPE})
+TARGET_C_PROPERTIES (ttsafe " " " ")
 TARGET_LINK_LIBRARIES (ttsafe ${HDF5_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
 SET_TARGET_PROPERTIES (ttsafe PROPERTIES FOLDER test)
 
@@ -482,6 +490,7 @@
 IF (HDF5_ENABLE_DEPRECATED_SYMBOLS)
   ADD_EXECUTABLE (err_compat ${HDF5_TEST_SOURCE_DIR}/err_compat.c)
   TARGET_NAMING (err_compat ${LIB_TYPE})
+  TARGET_C_PROPERTIES (err_compat " " " ")
   TARGET_LINK_LIBRARIES (err_compat ${HDF5_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
   SET_TARGET_PROPERTIES (err_compat PROPERTIES FOLDER test)
 
@@ -508,6 +517,7 @@
 #-- Adding test for error_test
 ADD_EXECUTABLE (error_test ${HDF5_TEST_SOURCE_DIR}/error_test.c)
 TARGET_NAMING (error_test ${LIB_TYPE})
+TARGET_C_PROPERTIES (error_test " " " ")
 TARGET_LINK_LIBRARIES (error_test ${HDF5_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
 SET_TARGET_PROPERTIES (error_test PROPERTIES FOLDER test)
 
@@ -534,6 +544,7 @@
 #-- Adding test for links_env
 ADD_EXECUTABLE (links_env ${HDF5_TEST_SOURCE_DIR}/links_env.c)
 TARGET_NAMING (links_env ${LIB_TYPE})
+TARGET_C_PROPERTIES (links_env " " " ")
 TARGET_LINK_LIBRARIES (links_env ${HDF5_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
 SET_TARGET_PROPERTIES (links_env PROPERTIES FOLDER test)
 
@@ -587,6 +598,7 @@
 
   ADD_EXECUTABLE (plugin ${HDF5_TEST_SOURCE_DIR}/plugin.c)
   TARGET_NAMING (plugin ${LIB_TYPE})
+  TARGET_C_PROPERTIES (plugin " " " ")
   TARGET_LINK_LIBRARIES (plugin ${HDF5_TEST_PLUGIN_LIB_TARGET})
   SET_TARGET_PROPERTIES (plugin PROPERTIES FOLDER test)
 
@@ -720,6 +732,7 @@
   MACRO (ADD_H5_GENERATOR genfile)
     ADD_EXECUTABLE (${genfile} ${HDF5_TEST_SOURCE_DIR}/${genfile}.c)
     TARGET_NAMING (${genfile} ${LIB_TYPE})
+    TARGET_C_PROPERTIES (${genfile} " " " ")
     TARGET_LINK_LIBRARIES (${genfile} ${HDF5_TEST_LIB_TARGET} ${HDF5_LIB_TARGET})
     SET_TARGET_PROPERTIES (${genfile} PROPERTIES FOLDER generator/test)
   ENDMACRO (ADD_H5_GENERATOR genfile)
Index: configure
===================================================================
--- configure	(revision 23770)
+++ configure	(revision 23771)
@@ -31025,8 +31025,8 @@
 ## Enable strict file format checks
 ##
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking Whether to perform strict file format checks" >&5
-$as_echo_n "checking Whether to perform strict file format checks... " >&6; };
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether to perform strict file format checks" >&5
+$as_echo_n "checking whether to perform strict file format checks... " >&6; };
 # Check whether --enable-strict-format-checks was given.
 if test "${enable_strict_format_checks+set}" = set; then :
   enableval=$enable_strict_format_checks; STRICT_CHECKS=$enableval
@@ -31062,8 +31062,8 @@
 ## ----------------------------------------------------------------------
 ## Enable embedded library information
 ##
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking Whether to have library information embedded in the executables" >&5
-$as_echo_n "checking Whether to have library information embedded in the executables... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking whether to have library information embedded in the executables" >&5
+$as_echo_n "checking whether to have library information embedded in the executables... " >&6; }
 # Check whether --enable-embedded-libinfo was given.
 if test "${enable_embedded_libinfo+set}" = set; then :
   enableval=$enable_embedded_libinfo; enable_embedded_libinfo=$enableval
Index: testpar/CMakeLists.txt
===================================================================
--- testpar/CMakeLists.txt	(revision 23770)
+++ testpar/CMakeLists.txt	(revision 23771)
@@ -28,6 +28,7 @@
 #-- Adding test for testhdf5
 ADD_EXECUTABLE (testphdf5 ${testphdf5_SRCS})
 TARGET_NAMING (testphdf5 ${LIB_TYPE})
+TARGET_C_PROPERTIES (testphdf5 " " " ")
 TARGET_LINK_LIBRARIES (testphdf5 ${HDF5_TEST_LIB_TARGET} ${HDF5_LIB_TARGET} ${LINK_LIBS})
 SET_TARGET_PROPERTIES (testphdf5 PROPERTIES FOLDER test/par)
 
@@ -36,6 +37,7 @@
 MACRO (ADD_H5P_TEST file)
   ADD_EXECUTABLE (${file} ${HDF5_TEST_PAR_SOURCE_DIR}/${file}.c)
   TARGET_NAMING (${file} ${LIB_TYPE})
+  TARGET_C_PROPERTIES (${file} " " " ")
   TARGET_LINK_LIBRARIES (${file} ${HDF5_TEST_LIB_TARGET} ${HDF5_LIB_TARGET} ${LINK_LIBS})
   SET_TARGET_PROPERTIES (${file} PROPERTIES FOLDER test/par)
 
Index: src/CMakeLists.txt
===================================================================
--- src/CMakeLists.txt	(revision 23770)
+++ src/CMakeLists.txt	(revision 23771)
@@ -614,6 +614,7 @@
 # specific type checks inside
 #-----------------------------------------------------------------------------
 ADD_EXECUTABLE (H5detect ${HDF5_SRC_DIR}/H5detect.c)
+TARGET_C_PROPERTIES (H5detect " " " ")
 IF (MSVC)
   TARGET_LINK_LIBRARIES (H5detect "ws2_32.lib")
 ENDIF (MSVC)
@@ -627,6 +628,7 @@
 )
 
 ADD_EXECUTABLE (H5make_libsettings ${HDF5_SRC_DIR}/H5make_libsettings.c)
+TARGET_C_PROPERTIES (H5make_libsettings " " " ")
 IF (MSVC)
   TARGET_LINK_LIBRARIES (H5make_libsettings "ws2_32.lib")
 ENDIF (MSVC)
@@ -676,6 +678,7 @@
 SET_SOURCE_FILES_PROPERTIES (${HDF5_BINARY_DIR}/H5overflow.h GENERATED)
 
 ADD_LIBRARY (${HDF5_LIB_TARGET} ${LIB_TYPE} ${common_SRCS} ${H5_PUBLIC_HEADERS} ${H5_PRIVATE_HEADERS})
+TARGET_C_PROPERTIES (${HDF5_LIB_TARGET} " " " ")
 TARGET_LINK_LIBRARIES (${HDF5_LIB_TARGET} ${LINK_LIBS})
 IF (NOT WIN32)
   TARGET_LINK_LIBRARIES (${HDF5_LIB_TARGET} dl)
Index: release_docs/USING_HDF5_CMake.txt
===================================================================
--- release_docs/USING_HDF5_CMake.txt	(revision 23770)
+++ release_docs/USING_HDF5_CMake.txt	(revision 23771)
@@ -197,6 +197,7 @@
 SET (example hdf_example)
 
 ADD_EXECUTABLE (${example} ${PROJECT_SOURCE_DIR}/${example}.c)
+TARGET_C_PROPERTIES (${example} " " " ")
 TARGET_LINK_LIBRARIES (${example} ${LINK_LIBS})
 
 ENABLE_TESTING ()
Index: tools/h5dump/testh5dump.sh.in
===================================================================
--- tools/h5dump/testh5dump.sh.in	(revision 23770)
+++ tools/h5dump/testh5dump.sh.in	(revision 23771)
@@ -59,7 +59,9 @@
 SRC_H5STAT_TESTFILES="$SRC_TOOLS/h5stat/testfiles"
 SRC_H5IMPORT_TESTFILES="$SRC_TOOLS/h5import/testfiles"
 
+TEST_P_DIR=./testfiles
 TESTDIR=./testfiles/std
+test -d $TEST_P_DIR || mkdir -p $TEST_P_DIR
 test -d $TESTDIR || mkdir -p $TESTDIR
 
 ######################################################################
@@ -272,6 +274,8 @@
 $SRC_H5DUMP_TESTFILES/tnullspace.ddl
 $SRC_H5DUMP_TESTFILES/trawdatafile.ddl
 $SRC_H5DUMP_TESTFILES/trawdatafile.exp
+$SRC_H5DUMP_TESTFILES/trawssetfile.ddl
+$SRC_H5DUMP_TESTFILES/trawssetfile.exp
 $SRC_H5DUMP_TESTFILES/zerodim.ddl
 $SRC_H5DUMP_TESTFILES/tordergr1.ddl
 $SRC_H5DUMP_TESTFILES/tordergr2.ddl
@@ -871,6 +875,7 @@
 TOOLTEST2 trawdatafile.exp --enable-error-stack -y -o trawdatafile.txt packedbits.h5
 TOOLTEST2 tnoddlfile.exp --enable-error-stack -O -y -o tnoddlfile.txt packedbits.h5
 TOOLTEST2A twithddlfile.exp twithddl.exp --enable-error-stack --ddl=twithddl.txt -y -o twithddlfile.txt packedbits.h5
+TOOLTEST2 trawssetfile.exp --enable-error-stack -d "/dset1[1,1;;;]" -y -o trawssetfile.txt tdset.h5
 
 # test for maximum display datasets
 TOOLTEST twidedisplay.ddl --enable-error-stack -w0 packedbits.h5
Index: tools/h5dump/testh5dumppbits.sh.in
===================================================================
--- tools/h5dump/testh5dumppbits.sh.in	(revision 23770)
+++ tools/h5dump/testh5dumppbits.sh.in	(revision 23771)
@@ -64,7 +64,9 @@
 SRC_H5STAT_TESTFILES="$SRC_TOOLS/h5stat/testfiles"
 SRC_H5IMPORT_TESTFILES="$SRC_TOOLS/h5import/testfiles"
 
+TEST_P_DIR=./testfiles
 TESTDIR=./testfiles/pbits
+test -d $TEST_P_DIR || mkdir -p $TEST_P_DIR
 test -d $TESTDIR || mkdir -p $TESTDIR
 
 ######################################################################
Index: tools/h5dump/testh5dumpxml.sh.in
===================================================================
--- tools/h5dump/testh5dumpxml.sh.in	(revision 23770)
+++ tools/h5dump/testh5dumpxml.sh.in	(revision 23771)
@@ -50,7 +50,9 @@
 SRC_H5STAT_TESTFILES="$SRC_TOOLS/h5stat/testfiles"
 SRC_H5IMPORT_TESTFILES="$SRC_TOOLS/h5import/testfiles"
 
+TEST_P_DIR=./testfiles
 TESTDIR=./testfiles/xml
+test -d $TEST_P_DIR || mkdir -p $TEST_P_DIR
 test -d $TESTDIR || mkdir -p $TESTDIR
 
 ######################################################################
Index: tools/h5dump/CMakeLists.txt
===================================================================
--- tools/h5dump/CMakeLists.txt	(revision 23770)
+++ tools/h5dump/CMakeLists.txt	(revision 23771)
@@ -15,6 +15,7 @@
     ${HDF5_TOOLS_H5DUMP_SOURCE_DIR}/h5dump_xml.c
 )
 TARGET_NAMING (h5dump ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5dump " " " ")
 TARGET_LINK_LIBRARIES (h5dump  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5dump PROPERTIES FOLDER tools)
 
@@ -33,6 +34,7 @@
   IF (HDF5_BUILD_GENERATORS AND NOT BUILD_SHARED_LIBS)
     ADD_EXECUTABLE (h5dumpgentest ${HDF5_TOOLS_H5DUMP_SOURCE_DIR}/h5dumpgentest.c)
     TARGET_NAMING (h5dumpgentest ${LIB_TYPE})
+    TARGET_C_PROPERTIES (h5dumpgentest " " " ")
     TARGET_LINK_LIBRARIES (h5dumpgentest ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
     SET_TARGET_PROPERTIES (h5dumpgentest PROPERTIES FOLDER generator/tools)
     
@@ -147,6 +149,7 @@
       ${HDF5_TOOLS_SRC_DIR}/testfiles/tno-subset.ddl
       ${HDF5_TOOLS_SRC_DIR}/testfiles/tnullspace.ddl
       ${HDF5_TOOLS_SRC_DIR}/testfiles/trawdatafile.ddl
+      ${HDF5_TOOLS_SRC_DIR}/testfiles/trawssetfile.ddl
       ${HDF5_TOOLS_SRC_DIR}/testfiles/zerodim.ddl
       ${HDF5_TOOLS_SRC_DIR}/testfiles/tordergr1.ddl
       ${HDF5_TOOLS_SRC_DIR}/testfiles/tordergr2.ddl
@@ -198,6 +201,7 @@
       tall-6.exp
       tnoddlfile.exp
       trawdatafile.exp
+      trawssetfile.exp
       tstr2bin2.exp
       tstr2bin6.exp
       twithddl.exp
@@ -1250,6 +1254,9 @@
           trawdatafile.out
           trawdatafile.out.err
           trawdatafile.txt
+          trawssetfile.out
+          trawssetfile.out.err
+          trawssetfile.txt
           tno-subset.out
           tno-subset.out.err
           tnullspace.out
@@ -1365,7 +1372,8 @@
   ADD_H5_TEST (tnoattrdata 0 --enable-error-stack -A -o tattr.h5)
   ADD_H5_TEST_EXPORT (trawdatafile packedbits.h5 0 --enable-error-stack -y -o)
   ADD_H5_TEST_EXPORT (tnoddlfile packedbits.h5 0 --enable-error-stack -O -y -o)
-
+  ADD_H5_TEST_EXPORT (trawssetfile tdset.h5 0 --enable-error-stack -d "/dset1[1,1;;;]" -y -o)
+  
   ADD_H5_TEST_EXPORT_DDL (twithddlfile packedbits.h5 0 twithddl --enable-error-stack --ddl=twithddl.txt -y -o)
   
   # test for maximum display datasets
Index: tools/h5repack/CMakeLists.txt
===================================================================
--- tools/h5repack/CMakeLists.txt	(revision 23770)
+++ tools/h5repack/CMakeLists.txt	(revision 23771)
@@ -22,6 +22,7 @@
 
 ADD_EXECUTABLE (h5repack ${REPACK_COMMON_SRCS} ${HDF5_TOOLS_H5REPACK_SOURCE_DIR}/h5repack_main.c)
 TARGET_NAMING (h5repack ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5repack " " " ")
 TARGET_LINK_LIBRARIES (h5repack  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5repack PROPERTIES FOLDER tools)
 
@@ -39,11 +40,13 @@
   # --------------------------------------------------------------------
   ADD_EXECUTABLE (testh5repack_detect_szip ${HDF5_TOOLS_H5REPACK_SOURCE_DIR}/testh5repack_detect_szip.c)
   TARGET_NAMING (testh5repack_detect_szip ${LIB_TYPE})
+  TARGET_C_PROPERTIES (testh5repack_detect_szip " " " ")
   TARGET_LINK_LIBRARIES (testh5repack_detect_szip ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
   SET_TARGET_PROPERTIES (testh5repack_detect_szip PROPERTIES FOLDER tools)
 
   ADD_EXECUTABLE (h5repacktest ${REPACK_COMMON_SRCS} ${HDF5_TOOLS_H5REPACK_SOURCE_DIR}/h5repacktst.c)
   TARGET_NAMING (h5repacktest ${LIB_TYPE})
+  TARGET_C_PROPERTIES (h5repacktest " " " ")
   TARGET_LINK_LIBRARIES (h5repacktest  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
   SET_TARGET_PROPERTIES (h5repacktest PROPERTIES FOLDER tools)
 
Index: tools/testfiles/trawssetfile.exp
===================================================================
--- tools/testfiles/trawssetfile.exp	(revision 0)
+++ tools/testfiles/trawssetfile.exp	(revision 23771)
@@ -0,0 +1,2 @@
+
+         2
\ No newline at end of file
Index: tools/testfiles/trawssetfile.ddl
===================================================================
--- tools/testfiles/trawssetfile.ddl	(revision 0)
+++ tools/testfiles/trawssetfile.ddl	(revision 23771)
@@ -0,0 +1,14 @@
+HDF5 "tdset.h5" {
+DATASET "/dset1" {
+   DATATYPE  H5T_STD_I32BE
+   DATASPACE  SIMPLE { ( 10, 20 ) / ( 10, 20 ) }
+   SUBSET {
+      START ( 1, 1 );
+      STRIDE ( 1, 1 );
+      COUNT ( 1, 1 );
+      BLOCK ( 1, 1 );
+      DATA {
+      }
+   }
+}
+}
Index: tools/h5jam/CMakeLists.txt
===================================================================
--- tools/h5jam/CMakeLists.txt	(revision 23770)
+++ tools/h5jam/CMakeLists.txt	(revision 23771)
@@ -11,21 +11,25 @@
 # --------------------------------------------------------------------
 ADD_EXECUTABLE (h5jam ${HDF5_TOOLS_H5JAM_SOURCE_DIR}/h5jam.c)
 TARGET_NAMING (h5jam ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5jam " " " ")
 TARGET_LINK_LIBRARIES (h5jam  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5jam PROPERTIES FOLDER tools)
 
 ADD_EXECUTABLE (getub ${HDF5_TOOLS_H5JAM_SOURCE_DIR}/getub.c)
 TARGET_NAMING (getub ${LIB_TYPE})
+TARGET_C_PROPERTIES (getub " " " ")
 TARGET_LINK_LIBRARIES (getub  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (getub PROPERTIES FOLDER tools)
 
 ADD_EXECUTABLE (tellub ${HDF5_TOOLS_H5JAM_SOURCE_DIR}/tellub.c)
 TARGET_NAMING (tellub ${LIB_TYPE})
+TARGET_C_PROPERTIES (tellub " " " ")
 TARGET_LINK_LIBRARIES (tellub  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (tellub PROPERTIES FOLDER tools)
 
 ADD_EXECUTABLE (h5unjam ${HDF5_TOOLS_H5JAM_SOURCE_DIR}/h5unjam.c)
 TARGET_NAMING (h5unjam ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5unjam " " " ")
 TARGET_LINK_LIBRARIES (h5unjam  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5unjam PROPERTIES FOLDER tools)
 
@@ -49,6 +53,7 @@
   IF (HDF5_BUILD_GENERATORS AND NOT BUILD_SHARED_LIBS)
     ADD_EXECUTABLE (h5jamgentest ${HDF5_TOOLS_H5JAM_SOURCE_DIR}/h5jamgentest.c)
     TARGET_NAMING (h5jamgentest ${LIB_TYPE})
+    TARGET_C_PROPERTIES (testhdf5 " " " ")
     TARGET_LINK_LIBRARIES (h5jamgentest ${HDF5_LIB_TARGET})
     SET_TARGET_PROPERTIES (h5jamgentest PROPERTIES FOLDER generator/tools)
     
Index: tools/h5diff/CMakeLists.txt
===================================================================
--- tools/h5diff/CMakeLists.txt	(revision 23770)
+++ tools/h5diff/CMakeLists.txt	(revision 23771)
@@ -14,6 +14,7 @@
     ${HDF5_TOOLS_H5DIFF_SOURCE_DIR}/h5diff_main.c
 )
 TARGET_NAMING (h5diff ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5diff " " " ")
 TARGET_LINK_LIBRARIES (h5diff  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5diff PROPERTIES FOLDER tools)
 
@@ -25,6 +26,7 @@
       ${HDF5_TOOLS_H5DIFF_SOURCE_DIR}/ph5diff_main.c
   )
   TARGET_NAMING (ph5diff ${LIB_TYPE})
+  TARGET_C_PROPERTIES (ph5diff " " " ")
   TARGET_LINK_LIBRARIES (ph5diff  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
   SET_TARGET_PROPERTIES (ph5diff PROPERTIES FOLDER tools)
 ENDIF (H5_HAVE_PARALLEL)
@@ -42,6 +44,7 @@
   IF (HDF5_BUILD_GENERATORS AND NOT BUILD_SHARED_LIBS)
     ADD_EXECUTABLE (h5diffgentest ${HDF5_TOOLS_H5DIFF_SOURCE_DIR}/h5diffgentest.c)
     TARGET_NAMING (h5diffgentest ${LIB_TYPE})
+    TARGET_C_PROPERTIES (h5diffgentest " " " ")
     TARGET_LINK_LIBRARIES (h5diffgentest ${HDF5_LIB_TARGET})
     SET_TARGET_PROPERTIES (h5diffgentest PROPERTIES FOLDER generator/tools)
     
Index: tools/lib/h5tools_dump.c
===================================================================
--- tools/lib/h5tools_dump.c	(revision 23770)
+++ tools/lib/h5tools_dump.c	(revision 23771)
@@ -314,13 +314,13 @@
                     if (region_space >= 0) {
                         if (h5tools_is_zero(memref, H5Tget_size(type))) {
                             ctx->need_prefix = TRUE;
-                            h5tools_simple_prefix(stream, info, ctx, curr_pos, 0);
+                            h5tools_simple_prefix(rawoutstream, info, ctx, curr_pos, 0);
 
                             /* Render the region element begin */
                             h5tools_str_reset(&buffer);
                             h5tools_str_append(&buffer, "NULL");
 
-                            dimension_break = h5tools_render_element(stream, info,
+                            dimension_break = h5tools_render_element(rawoutstream, info,
                                        ctx, &buffer, &curr_pos, ncols, i, elmt_counter);
                         }
                         else {
@@ -328,25 +328,25 @@
                                 HERROR(H5E_tools_g, H5E_tools_min_id_g, "H5Rget_name failed");
 
                             ctx->need_prefix = TRUE;
-                            h5tools_simple_prefix(stream, info, ctx, curr_pos+i, 0);
+                            h5tools_simple_prefix(rawoutstream, info, ctx, curr_pos+i, 0);
 
                             /* Render the region element begin */
                             h5tools_str_reset(&buffer);
                             h5tools_str_append(&buffer, info->dset_format, ref_name);
 
-                            dimension_break = h5tools_render_element(stream, info,
+                            dimension_break = h5tools_render_element(rawoutstream, info,
                                        ctx, &buffer, &curr_pos, ncols, i, elmt_counter);
 
                             region_type = H5Sget_select_type(region_space);
                             if(region_type == H5S_SEL_POINTS)
                                 /* Print point information */
                                 dimension_break = h5tools_dump_region_data_points(
-                                                       region_space, region_id, stream, info, ctx,
+                                                       region_space, region_id, rawoutstream, info, ctx,
                                                        &buffer, &curr_pos, ncols, i, elmt_counter);
                             else if(region_type == H5S_SEL_HYPERSLABS)
                                 /* Print block information */
                                 dimension_break = h5tools_dump_region_data_blocks(
-                                                       region_space, region_id, stream, info, ctx,
+                                                       region_space, region_id, rawoutstream, info, ctx,
                                                        &buffer, &curr_pos, ncols, i, elmt_counter);
                             else
                                 HERROR(H5E_tools_g, H5E_tools_min_id_g, "invalid region type");
@@ -1790,7 +1790,7 @@
         if(!sset)
             status = h5tools_dump_simple_dset(rawdatastream, info, ctx, dset, p_type);
         else
-            status = h5tools_dump_simple_subset(stream, info, ctx, dset, p_type, sset);
+            status = h5tools_dump_simple_subset(rawdatastream, info, ctx, dset, p_type, sset);
     }
     else
         /* space is H5S_NULL */
Index: tools/lib/CMakeLists.txt
===================================================================
--- tools/lib/CMakeLists.txt	(revision 23770)
+++ tools/lib/CMakeLists.txt	(revision 23771)
@@ -37,8 +37,9 @@
 )
 
 ADD_LIBRARY (${HDF5_TOOLS_LIB_TARGET} ${LIB_TYPE} ${H5_TOOLS_LIB_SRCS} ${H5_TOOLS_LIB_HDRS})
+TARGET_C_PROPERTIES (${HDF5_TOOLS_LIB_TARGET} " " " ")
 TARGET_LINK_LIBRARIES (${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
-SET_GLOBAL_VARIABLE( HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_TOOLS_LIB_TARGET}")
+SET_GLOBAL_VARIABLE (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_TOOLS_LIB_TARGET}")
 H5_SET_LIB_OPTIONS (
     ${HDF5_TOOLS_LIB_TARGET} ${HDF5_TOOLS_LIB_NAME}
     ${LIB_TYPE}
Index: tools/h5copy/CMakeLists.txt
===================================================================
--- tools/h5copy/CMakeLists.txt	(revision 23770)
+++ tools/h5copy/CMakeLists.txt	(revision 23771)
@@ -11,6 +11,7 @@
 # --------------------------------------------------------------------
 ADD_EXECUTABLE (h5copy ${HDF5_TOOLS_H5COPY_SOURCE_DIR}/h5copy.c)
 TARGET_NAMING (h5copy ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5copy " " " ")
 TARGET_LINK_LIBRARIES (h5copy  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5copy PROPERTIES FOLDER tools)
 
@@ -26,6 +27,7 @@
   IF (HDF5_BUILD_GENERATORS AND NOT BUILD_SHARED_LIBS)
     ADD_EXECUTABLE (h5copygentest ${HDF5_TOOLS_H5COPY_SOURCE_DIR}/h5copygentest.c)
     TARGET_NAMING (h5copygentest ${LIB_TYPE})
+    TARGET_C_PROPERTIES (h5copygentest " " " ")
     TARGET_LINK_LIBRARIES (h5copygentest ${HDF5_LIB_TARGET})
     SET_TARGET_PROPERTIES (h5copygentest PROPERTIES FOLDER generator/tools)
 
Index: tools/h5import/CMakeLists.txt
===================================================================
--- tools/h5import/CMakeLists.txt	(revision 23770)
+++ tools/h5import/CMakeLists.txt	(revision 23771)
@@ -11,6 +11,7 @@
 # --------------------------------------------------------------------
 ADD_EXECUTABLE (h5import ${HDF5_TOOLS_H5IMPORT_SOURCE_DIR}/h5import.c)
 TARGET_NAMING (h5import ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5import " " " ")
 TARGET_LINK_LIBRARIES (h5import  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 #SET_TARGET_PROPERTIES (h5import PROPERTIES COMPILE_DEFINITIONS H5DEBUGIMPORT)
 SET_TARGET_PROPERTIES (h5import PROPERTIES FOLDER tools)
@@ -29,6 +30,7 @@
   # --------------------------------------------------------------------
   ADD_EXECUTABLE (h5importtest ${HDF5_TOOLS_H5IMPORT_SOURCE_DIR}/h5importtest.c)
   TARGET_NAMING (h5importtest ${LIB_TYPE})
+  TARGET_C_PROPERTIES (h5importtest " " " ")
   TARGET_LINK_LIBRARIES (h5importtest ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
   SET_TARGET_PROPERTIES (h5importtest PROPERTIES FOLDER tools)
 
Index: tools/h5stat/CMakeLists.txt
===================================================================
--- tools/h5stat/CMakeLists.txt	(revision 23770)
+++ tools/h5stat/CMakeLists.txt	(revision 23771)
@@ -11,6 +11,7 @@
 # --------------------------------------------------------------------
 ADD_EXECUTABLE (h5stat ${HDF5_TOOLS_H5STAT_SOURCE_DIR}/h5stat.c)
 TARGET_NAMING (h5stat ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5stat " " " ")
 TARGET_LINK_LIBRARIES (h5stat  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5stat PROPERTIES FOLDER tools)
 
@@ -29,6 +30,7 @@
   IF (HDF5_BUILD_GENERATORS AND NOT BUILD_SHARED_LIBS)
     ADD_EXECUTABLE (h5stat_gentest ${HDF5_TOOLS_H5STAT_SOURCE_DIR}/h5stat_gentest.c)
     TARGET_NAMING (h5stat_gentest ${LIB_TYPE})
+    TARGET_C_PROPERTIES (h5stat_gentest " " " ")
     TARGET_LINK_LIBRARIES (h5stat_gentest ${HDF5_LIB_TARGET})
     SET_TARGET_PROPERTIES (h5stat_gentest PROPERTIES FOLDER generator/tools)
     
Index: tools/h5ls/CMakeLists.txt
===================================================================
--- tools/h5ls/CMakeLists.txt	(revision 23770)
+++ tools/h5ls/CMakeLists.txt	(revision 23771)
@@ -11,6 +11,7 @@
 #-----------------------------------------------------------------------------
 ADD_EXECUTABLE (h5ls ${HDF5_TOOLS_H5LS_SOURCE_DIR}/h5ls.c)
 TARGET_NAMING (h5ls ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5ls " " " ")
 TARGET_LINK_LIBRARIES (h5ls  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5ls PROPERTIES FOLDER tools)
 
Index: tools/misc/CMakeLists.txt
===================================================================
--- tools/misc/CMakeLists.txt	(revision 23770)
+++ tools/misc/CMakeLists.txt	(revision 23771)
@@ -12,16 +12,19 @@
 #-- Misc Executables
 ADD_EXECUTABLE (h5debug ${HDF5_TOOLS_MISC_SOURCE_DIR}/h5debug.c)
 TARGET_NAMING (h5debug ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5debug " " " ")
 TARGET_LINK_LIBRARIES (h5debug ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5debug PROPERTIES FOLDER tools)
 
 ADD_EXECUTABLE (h5repart ${HDF5_TOOLS_MISC_SOURCE_DIR}/h5repart.c)
 TARGET_NAMING (h5repart ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5repart " " " ")
 TARGET_LINK_LIBRARIES (h5repart ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5repart PROPERTIES FOLDER tools)
 
 ADD_EXECUTABLE (h5mkgrp ${HDF5_TOOLS_MISC_SOURCE_DIR}/h5mkgrp.c)
 TARGET_NAMING (h5mkgrp ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5mkgrp " " " ")
 TARGET_LINK_LIBRARIES (h5mkgrp  ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5mkgrp PROPERTIES FOLDER tools)
 
@@ -44,6 +47,7 @@
   IF (HDF5_BUILD_GENERATORS AND NOT BUILD_SHARED_LIBS)
     ADD_EXECUTABLE (h5repart_gentest ${HDF5_TOOLS_MISC_SOURCE_DIR}/h5repart_gentest.c)
     TARGET_NAMING (h5repart_gentest ${LIB_TYPE})
+    TARGET_C_PROPERTIES (h5repart_gentest " " " ")
     TARGET_LINK_LIBRARIES (h5repart_gentest ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
     SET_TARGET_PROPERTIES (h5repart_gentest PROPERTIES FOLDER generator/tools)
     #ADD_TEST (NAME h5repart_gentest COMMAND $<TARGET_FILE:h5repart_gentest>)
@@ -51,6 +55,7 @@
 
   ADD_EXECUTABLE (h5repart_test ${HDF5_TOOLS_MISC_SOURCE_DIR}/repart_test.c)
   TARGET_NAMING (h5repart_test ${LIB_TYPE})
+  TARGET_C_PROPERTIES (h5repart_test " " " ")
   TARGET_LINK_LIBRARIES (h5repart_test ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
   SET_TARGET_PROPERTIES (h5repart_test PROPERTIES FOLDER tools)
   
Index: UserMacros.cmake
===================================================================
--- UserMacros.cmake	(revision 0)
+++ UserMacros.cmake	(revision 23771)
@@ -0,0 +1,17 @@
+########################################################
+#  Include file for user options
+########################################################
+
+#-----------------------------------------------------------------------------
+# Option to Build with User Defined Values
+#-----------------------------------------------------------------------------
+MACRO (MACRO_USER_DEFINED_LIBS)
+  SET (USER_DEFINED_VALUE "FALSE")
+ENDMACRO (MACRO_USER_DEFINED_LIBS)
+
+#-------------------------------------------------------------------------------
+OPTION (BUILD_USER_DEFINED_LIBS "Build With User Defined Values" OFF)
+IF (BUILD_USER_DEFINED_LIBS)
+  MACRO_USER_DEFINED_LIBS ()
+ENDIF (BUILD_USER_DEFINED_LIBS)
+ 
\ No newline at end of file
Index: hl/test/CMakeLists.txt
===================================================================
--- hl/test/CMakeLists.txt	(revision 23770)
+++ hl/test/CMakeLists.txt	(revision 23771)
@@ -23,6 +23,7 @@
 MACRO (HL_ADD_TEST hl_name files)
   ADD_EXECUTABLE (hl_${hl_name} ${hl_name}.c)
   TARGET_NAMING (hl_${hl_name} ${LIB_TYPE})
+  TARGET_C_PROPERTIES (hl_${hl_name} " " " ")
   TARGET_LINK_LIBRARIES (hl_${hl_name}
       ${HDF5_HL_LIB_TARGET}
       ${HDF5_LIB_TARGET}
@@ -97,6 +98,7 @@
 IF (HDF5_BUILD_GENERATORS AND NOT BUILD_SHARED_LIBS)
   ADD_EXECUTABLE (hl_gen_test_ds gen_test_ds.c)
   TARGET_NAMING (hl_gen_test_ds ${LIB_TYPE})
+  TARGET_C_PROPERTIES (hl_gen_test_ds " " " ")
   TARGET_LINK_LIBRARIES (hl_gen_test_ds
       ${HDF5_HL_LIB_TARGET}
       ${HDF5_LIB_TARGET}
Index: hl/tools/CMakeLists.txt
===================================================================
--- hl/tools/CMakeLists.txt	(revision 23770)
+++ hl/tools/CMakeLists.txt	(revision 23771)
@@ -19,6 +19,7 @@
 
 ADD_EXECUTABLE (gif2h5 ${GIF2H5_SRCS})
 TARGET_NAMING (gif2h5 ${LIB_TYPE})
+TARGET_C_PROPERTIES (gif2h5 " " " ")
 TARGET_LINK_LIBRARIES (gif2h5 ${HDF5_HL_LIB_TARGET} ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
 SET_TARGET_PROPERTIES (gif2h5 PROPERTIES FOLDER tools/hl)
 
@@ -29,6 +30,7 @@
 )
 ADD_EXECUTABLE (h52gif ${hdf2gif_SRCS})
 TARGET_NAMING (h52gif ${LIB_TYPE})
+TARGET_C_PROPERTIES (h52gif " " " ")
 TARGET_LINK_LIBRARIES (h52gif ${HDF5_HL_LIB_TARGET} ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
 SET_TARGET_PROPERTIES (h52gif PROPERTIES FOLDER tools/hl)
 
@@ -40,6 +42,7 @@
 IF (HDF5_BUILD_GENERATORS AND NOT BUILD_SHARED_LIBS)
     ADD_EXECUTABLE (hl_h52gifgentest ${HDF5_HL_TOOLS_SOURCE_DIR}/gif2h5/h52gifgentst.c)
     TARGET_NAMING (hl_h52gifgentest ${LIB_TYPE})
+    TARGET_C_PROPERTIES (hl_h52gifgentest " " " ")
     TARGET_LINK_LIBRARIES (hl_h52gifgentest ${HDF5_HL_LIB_TARGET} ${HDF5_LIB_TARGET})
     SET_TARGET_PROPERTIES (hl_h52gifgentest PROPERTIES FOLDER generator/tools/hl)
     
Index: hl/src/CMakeLists.txt
===================================================================
--- hl/src/CMakeLists.txt	(revision 23770)
+++ hl/src/CMakeLists.txt	(revision 23771)
@@ -36,6 +36,7 @@
 )
 
 ADD_LIBRARY (${HDF5_HL_LIB_TARGET} ${LIB_TYPE} ${HL_SRCS} ${HL_HEADERS})
+TARGET_C_PROPERTIES (${HDF5_HL_LIB_TARGET} " " " ")
 TARGET_LINK_LIBRARIES (${HDF5_HL_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_GLOBAL_VARIABLE (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_HL_LIB_TARGET}")
 H5_SET_LIB_OPTIONS (${HDF5_HL_LIB_TARGET} ${HDF5_HL_LIB_NAME} ${LIB_TYPE})
Index: hl/c++/test/CMakeLists.txt
===================================================================
--- hl/c++/test/CMakeLists.txt	(revision 23770)
+++ hl/c++/test/CMakeLists.txt	(revision 23771)
@@ -17,6 +17,7 @@
   INCLUDE_DIRECTORIES (${HDF5_CPP_SRC_DIR}/src)
 
   ADD_EXECUTABLE (hl_ptableTest ${HDF5_HL_CPP_TEST_SOURCE_DIR}/ptableTest.cpp)
+  TARGET_C_PROPERTIES (hl_ptableTest " " " ")
   TARGET_NAMING (hl_ptableTest ${LIB_TYPE})
   TARGET_LINK_LIBRARIES (
       hl_ptableTest
Index: hl/c++/src/CMakeLists.txt
===================================================================
--- hl/c++/src/CMakeLists.txt	(revision 23770)
+++ hl/c++/src/CMakeLists.txt	(revision 23771)
@@ -10,7 +10,8 @@
 SET (HDF5_HL_CPP_SRCS ${HDF5_HL_CPP_SRC_SOURCE_DIR}/H5PacketTable.cpp)
 SET (HDF5_HL_CPP_HDRS ${HDF5_HL_CPP_SRC_SOURCE_DIR}/H5PacketTable.h)
 
-ADD_LIBRARY ( ${HDF5_HL_CPP_LIB_TARGET} ${LIB_TYPE} ${HDF5_HL_CPP_SRCS})
+ADD_LIBRARY (${HDF5_HL_CPP_LIB_TARGET} ${LIB_TYPE} ${HDF5_HL_CPP_SRCS})
+TARGET_C_PROPERTIES (${HDF5_HL_CPP_LIB_TARGET} " " " ")
 TARGET_LINK_LIBRARIES (
     ${HDF5_HL_CPP_LIB_TARGET}
     ${HDF5_HL_LIB_TARGET}
Index: hl/c++/examples/CMakeLists.txt
===================================================================
--- hl/c++/examples/CMakeLists.txt	(revision 23770)
+++ hl/c++/examples/CMakeLists.txt	(revision 23771)
@@ -12,6 +12,7 @@
 # --------------------------------------------------------------------
 ADD_EXECUTABLE (ptExampleFL ${HDF5_HL_CPP_EXAMPLES_SOURCE_DIR}/ptExampleFL.cpp)
 TARGET_NAMING (ptExampleFL ${LIB_TYPE})
+TARGET_C_PROPERTIES (ptExampleFL " " " ")
 TARGET_LINK_LIBRARIES (
     ptExampleFL
     ${HDF5_HL_CPP_LIB_TARGET}
Index: hl/fortran/test/CMakeLists.txt
===================================================================
--- hl/fortran/test/CMakeLists.txt	(revision 23770)
+++ hl/fortran/test/CMakeLists.txt	(revision 23771)
@@ -23,7 +23,7 @@
 #-- Adding test for hl_f90_tstds
 ADD_EXECUTABLE (hl_f90_tstds tstds.f90)
 TARGET_NAMING (hl_f90_tstds ${LIB_TYPE})
-TARGET_FORTRAN_WIN_PROPERTIES (hl_f90_tstds "")
+TARGET_FORTRAN_PROPERTIES (hl_f90_tstds " " " ")
 TARGET_LINK_LIBRARIES (hl_f90_tstds ${HDF5_HL_F90_LIB_TARGET} ${HDF5_F90_LIB_TARGET})
 SET_TARGET_PROPERTIES (hl_f90_tstds PROPERTIES LINKER_LANGUAGE Fortran)
 SET_TARGET_PROPERTIES (hl_f90_tstds PROPERTIES FOLDER test/hl/fortran)
@@ -33,7 +33,7 @@
 #-- Adding test for hl_f90_tstlite
 ADD_EXECUTABLE (hl_f90_tstlite tstlite.f90)
 TARGET_NAMING (hl_f90_tstlite ${LIB_TYPE})
-TARGET_FORTRAN_WIN_PROPERTIES (hl_f90_tstlite "")
+TARGET_FORTRAN_PROPERTIES (hl_f90_tstlite " " " ")
 TARGET_LINK_LIBRARIES (hl_f90_tstlite ${HDF5_HL_F90_LIB_TARGET} ${HDF5_F90_LIB_TARGET})
 SET_TARGET_PROPERTIES (hl_f90_tstlite PROPERTIES LINKER_LANGUAGE Fortran)
 SET_TARGET_PROPERTIES (hl_f90_tstlite PROPERTIES FOLDER test/hl/fortran)
@@ -43,7 +43,7 @@
 #-- Adding test for hl_f90_tstimage
 ADD_EXECUTABLE (hl_f90_tstimage tstimage.f90)
 TARGET_NAMING (hl_f90_tstimage ${LIB_TYPE})
-TARGET_FORTRAN_WIN_PROPERTIES (hl_f90_tstimage "")
+TARGET_FORTRAN_PROPERTIES (hl_f90_tstimage " " " ")
 TARGET_LINK_LIBRARIES (hl_f90_tstimage  ${HDF5_HL_F90_LIB_TARGET} ${HDF5_F90_LIB_TARGET})
 SET_TARGET_PROPERTIES (hl_f90_tstimage PROPERTIES LINKER_LANGUAGE Fortran)
 SET_TARGET_PROPERTIES (hl_f90_tstimage PROPERTIES FOLDER test/hl/fortran)
@@ -53,7 +53,7 @@
 #-- Adding test for hl_f90_tsttable
 ADD_EXECUTABLE (hl_f90_tsttable tsttable.f90)
 TARGET_NAMING (hl_f90_tsttable ${LIB_TYPE})
-TARGET_FORTRAN_WIN_PROPERTIES (hl_f90_tsttable "")
+TARGET_FORTRAN_PROPERTIES (hl_f90_tsttable " " " ")
 TARGET_LINK_LIBRARIES (hl_f90_tsttable ${HDF5_HL_F90_LIB_TARGET} ${HDF5_F90_LIB_TARGET})
 SET_TARGET_PROPERTIES (hl_f90_tsttable PROPERTIES LINKER_LANGUAGE Fortran)
 SET_TARGET_PROPERTIES (hl_f90_tsttable PROPERTIES FOLDER test/hl/fortran)
Index: hl/fortran/src/CMakeLists.txt
===================================================================
--- hl/fortran/src/CMakeLists.txt	(revision 23770)
+++ hl/fortran/src/CMakeLists.txt	(revision 23771)
@@ -26,6 +26,7 @@
 SET (HDF5_HL_F90_HEADERS ${HDF5_HL_F90_SRC_SOURCE_DIR}/H5LTf90proto.h)
 
 ADD_LIBRARY (${HDF5_HL_F90_C_LIB_TARGET} ${LIB_TYPE} ${HDF5_HL_F90_C_SRCS} ${HDF5_HL_F90_HEADERS})
+TARGET_C_PROPERTIES (${HDF5_HL_F90_C_LIB_TARGET} " " " ")
 TARGET_LINK_LIBRARIES (${HDF5_HL_F90_C_LIB_TARGET} ${HDF5_F90_C_LIB_TARGET} ${HDF5_HL_LIB_TARGET})
 SET_GLOBAL_VARIABLE (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_HL_F90_C_LIB_TARGET}")
 H5_SET_LIB_OPTIONS (${HDF5_HL_F90_C_LIB_TARGET} ${HDF5_HL_F90_C_LIB_NAME} ${LIB_TYPE})
@@ -61,7 +62,7 @@
         HDF5F90_WINDOWS
   )
 ENDIF (WIN32 AND NOT CYGWIN)
-TARGET_FORTRAN_WIN_PROPERTIES (${HDF5_HL_F90_LIB_TARGET} ${SHARED_LINK_FLAGS})
+TARGET_FORTRAN_PROPERTIES (${HDF5_HL_F90_LIB_TARGET} " " ${SHARED_LINK_FLAGS})
 SET_TARGET_PROPERTIES (${HDF5_HL_F90_LIB_TARGET} PROPERTIES LINKER_LANGUAGE Fortran)
 TARGET_LINK_LIBRARIES (${HDF5_HL_F90_LIB_TARGET} ${HDF5_HL_F90_C_LIB_TARGET} ${HDF5_F90_LIB_TARGET})
 SET_GLOBAL_VARIABLE (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_HL_F90_LIB_TARGET}")
Index: hl/fortran/examples/CMakeLists.txt
===================================================================
--- hl/fortran/examples/CMakeLists.txt	(revision 23770)
+++ hl/fortran/examples/CMakeLists.txt	(revision 23771)
@@ -18,7 +18,7 @@
 FOREACH (example ${examples})
   ADD_EXECUTABLE (hl_f90_ex_${example} ${HDF5_HL_F90_EXAMPLES_SOURCE_DIR}/${example}.f90)
   TARGET_NAMING (hl_f90_ex_${example} ${LIB_TYPE})
-  TARGET_FORTRAN_WIN_PROPERTIES (hl_f90_ex_${example} "")
+  TARGET_FORTRAN_PROPERTIES (hl_f90_ex_${example} " " " ")
   TARGET_LINK_LIBRARIES (hl_f90_ex_${example} 
       ${HDF5_HL_F90_LIB_TARGET}
       ${HDF5_F90_LIB_TARGET}
Index: hl/examples/CMakeLists.txt
===================================================================
--- hl/examples/CMakeLists.txt	(revision 23770)
+++ hl/examples/CMakeLists.txt	(revision 23771)
@@ -43,6 +43,7 @@
 FOREACH (example ${examples})
   ADD_EXECUTABLE (hl_ex_${example} ${HDF5_HL_EXAMPLES_SOURCE_DIR}/${example}.c)
   TARGET_NAMING (hl_ex_${example} ${LIB_TYPE})
+  TARGET_C_PROPERTIES (hl_ex_${example} " " " ")
   TARGET_LINK_LIBRARIES (hl_ex_${example} ${HDF5_HL_LIB_TARGET} ${HDF5_LIB_TARGET})
   SET_TARGET_PROPERTIES (hl_ex_${example} PROPERTIES FOLDER examples/hl)
 
Index: configure.ac
===================================================================
--- configure.ac	(revision 23770)
+++ configure.ac	(revision 23771)
@@ -4305,7 +4305,7 @@
 ## Enable strict file format checks
 ##
 AC_SUBST([STRICT_FORMAT_CHECKS])
-AC_MSG_CHECKING([Whether to perform strict file format checks]);
+AC_MSG_CHECKING([whether to perform strict file format checks]);
 AC_ARG_ENABLE([strict-format-checks],
               [AS_HELP_STRING([--enable-strict-format-checks],
                      [Enable strict file format checks, default=yes if
@@ -4338,7 +4338,7 @@
 ## ----------------------------------------------------------------------
 ## Enable embedded library information
 ##
-AC_MSG_CHECKING([Whether to have library information embedded in the executables])
+AC_MSG_CHECKING([whether to have library information embedded in the executables])
 AC_ARG_ENABLE([embedded-libinfo],
     [AS_HELP_STRING([--enable-embedded-libinfo],
 	[Enable embedded library information [default=yes]])],
Index: config/cmake/ConfigureChecks.cmake
===================================================================
--- config/cmake/ConfigureChecks.cmake	(revision 23770)
+++ config/cmake/ConfigureChecks.cmake	(revision 23771)
@@ -183,15 +183,15 @@
   ENDIF (MINGW)
   SET (H5_HAVE_LIBWS2_32 1)
   SET (H5_HAVE_LIBWSOCK32 1)
+
+  #-----------------------------------------------------------------------------
+  # These tests need to be manually SET for windows since there is currently
+  # something not quite correct with the actual test implementation. This affects
+  # the 'dt_arith' test and most likely lots of other code
+  # ----------------------------------------------------------------------------
+  SET (H5_FP_TO_ULLONG_RIGHT_MAXIMUM "" CACHE INTERNAL "")
 ENDIF (WINDOWS)
 
-#-----------------------------------------------------------------------------
-# These tests need to be manually SET for windows since there is currently
-# something not quite correct with the actual test implementation. This affects
-# the 'dt_arith' test and most likely lots of other code
-# ----------------------------------------------------------------------------
-SET (H5_FP_TO_ULLONG_RIGHT_MAXIMUM "" CACHE INTERNAL "")
-
 # ----------------------------------------------------------------------
 # END of WINDOWS Hard code Values
 # ----------------------------------------------------------------------
@@ -1029,7 +1029,9 @@
 # integers except 'unsigned long long'.  Other HP-UX systems are unknown
 # yet. (1/8/05 - SLU)
 #
-H5ConversionTests (H5_LDOUBLE_TO_INTEGER_WORKS "Checking IF converting from long double to integers works")
+IF (NOT MSVC)
+  H5ConversionTests (H5_LDOUBLE_TO_INTEGER_WORKS "Checking IF converting from long double to integers works")
+ENDIF (NOT MSVC)
 # -----------------------------------------------------------------------
 # Set flag to indicate that the machine can handle conversion from
 # integers to long double.  (This flag should be set "yes" for all
@@ -1103,7 +1105,9 @@
 # where the last 2 bytes of mantissa are lost when compiler tries to do
 # the conversion, and Cygwin where compiler doesn't do rounding correctly.)
 #
-H5ConversionTests (H5_ULLONG_TO_LDOUBLE_PRECISION "Checking IF converting unsigned long long to long double with precision")
+IF (NOT MSVC)
+  H5ConversionTests (H5_ULLONG_TO_LDOUBLE_PRECISION "Checking IF converting unsigned long long to long double with precision")
+ENDIF (NOT MSVC)
 # ----------------------------------------------------------------------
 # Set the flag to indicate that the machine can handle overflow converting
 # all floating-point to all integer types.
Index: config/cmake/HDFMacros.cmake
===================================================================
--- config/cmake/HDFMacros.cmake	(revision 23770)
+++ config/cmake/HDFMacros.cmake	(revision 23771)
@@ -121,20 +121,67 @@
 ENDMACRO (HDF_SET_LIB_OPTIONS)
 
 #-------------------------------------------------------------------------------
-MACRO (TARGET_FORTRAN_WIN_PROPERTIES forttarget addlinkflags)
-  IF (WIN32 AND MSVC)
+MACRO (TARGET_C_PROPERTIES wintarget addcompileflags addlinkflags)
+  IF (MSVC)
+    TARGET_MSVC_PROPERTIES (${wintarget} "${addcompileflags} ${WIN_COMPILE_FLAGS}" "${addlinkflags} ${WIN_LINK_FLAGS}")
+  ELSE (MSVC)
     IF (BUILD_SHARED_LIBS)
+      SET_TARGET_PROPERTIES (${wintarget}
+          PROPERTIES
+              COMPILE_FLAGS "${addcompileflags}"
+              LINK_FLAGS "${addlinkflags}"
+      ) 
+    ELSE (BUILD_SHARED_LIBS)
+      SET_TARGET_PROPERTIES (${wintarget}
+          PROPERTIES
+              COMPILE_FLAGS "${addcompileflags}"
+              LINK_FLAGS "${addlinkflags}"
+      ) 
+    ENDIF (BUILD_SHARED_LIBS)
+  ENDIF (MSVC)
+ENDMACRO (TARGET_C_PROPERTIES)
+
+#-------------------------------------------------------------------------------
+MACRO (TARGET_MSVC_PROPERTIES wintarget addcompileflags addlinkflags)
+  IF (MSVC)
+    IF (BUILD_SHARED_LIBS)
+      SET_TARGET_PROPERTIES (${wintarget}
+          PROPERTIES
+              COMPILE_FLAGS "/dll ${addcompileflags}"
+              LINK_FLAGS "/SUBSYSTEM:CONSOLE ${addlinkflags}"
+      ) 
+    ELSE (BUILD_SHARED_LIBS)
+      SET_TARGET_PROPERTIES (${wintarget}
+          PROPERTIES
+              COMPILE_FLAGS "${addcompileflags}"
+              LINK_FLAGS "/SUBSYSTEM:CONSOLE ${addlinkflags}"
+      ) 
+    ENDIF (BUILD_SHARED_LIBS)
+  ENDIF (MSVC)
+ENDMACRO (TARGET_MSVC_PROPERTIES)
+
+#-------------------------------------------------------------------------------
+MACRO (TARGET_FORTRAN_PROPERTIES forttarget addcompileflags addlinkflags)
+  IF (WIN32)
+    TARGET_FORTRAN_WIN_PROPERTIES (${forttarget} "${addcompileflags} ${WIN_COMPILE_FLAGS}" "${addlinkflags} ${WIN_LINK_FLAGS}")
+  ENDIF (WIN32)
+ENDMACRO (TARGET_FORTRAN_PROPERTIES)
+
+#-------------------------------------------------------------------------------
+MACRO (TARGET_FORTRAN_WIN_PROPERTIES forttarget addcompileflags addlinkflags)
+  IF (MSVC)
+    IF (BUILD_SHARED_LIBS)
       SET_TARGET_PROPERTIES (${forttarget}
           PROPERTIES
-              COMPILE_FLAGS "/dll"
+              COMPILE_FLAGS "/dll ${addcompileflags}"
               LINK_FLAGS "/SUBSYSTEM:CONSOLE ${addlinkflags}"
       ) 
     ELSE (BUILD_SHARED_LIBS)
       SET_TARGET_PROPERTIES (${forttarget}
           PROPERTIES
-              COMPILE_FLAGS "/MD"
+              COMPILE_FLAGS "${addcompileflags}"
               LINK_FLAGS "/SUBSYSTEM:CONSOLE ${addlinkflags}"
       ) 
     ENDIF (BUILD_SHARED_LIBS)
-  ENDIF (WIN32 AND MSVC)
+  ENDIF (MSVC)
 ENDMACRO (TARGET_FORTRAN_WIN_PROPERTIES)
Index: config/cmake/UserMacros/Windows_MT.cmake
===================================================================
--- config/cmake/UserMacros/Windows_MT.cmake	(revision 0)
+++ config/cmake/UserMacros/Windows_MT.cmake	(revision 23771)
@@ -0,0 +1,39 @@
+########################################################
+#  Include file for user options
+########################################################
+
+# To use this option, copy both the macro and option code
+# into the root UserMacros.cmake file. 
+
+#-----------------------------------------------------------------------------
+# Option to Build with Static CRT libraries on Windows
+#-------------------------------------------------------------------------------
+MACRO (TARGET_STATIC_CRT_FLAGS)
+  IF (MSVC AND NOT BUILD_SHARED_LIBS)
+    FOREACH (flag_var
+        CMAKE_C_FLAGS CMAKE_C_FLAGS_DEBUG CMAKE_C_FLAGS_RELEASE
+        CMAKE_C_FLAGS_MINSIZEREL CMAKE_C_FLAGS_RELWITHDEBINFO
+        CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
+        CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+      IF (${flag_var} MATCHES "/MD")
+        STRING (REGEX REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
+      ENDIF (${flag_var} MATCHES "/MD")
+    ENDFOREACH (flag_var)
+    FOREACH (flag_var
+        CMAKE_Fortran_FLAGS CMAKE_Fortran_FLAGS_DEBUG CMAKE_Fortran_FLAGS_RELEASE
+        CMAKE_Fortran_FLAGS_MINSIZEREL CMAKE_Fortran_FLAGS_RELWITHDEBINFO)
+      IF (${flag_var} MATCHES "/libs:dll")
+        STRING (REGEX REPLACE "/libs:dll" "/libs:static" ${flag_var} "${${flag_var}}")
+      ENDIF (${flag_var} MATCHES "/libs:dll")
+    ENDFOREACH (flag_var)
+    SET (WIN_COMPILE_FLAGS "/MT")
+    SET (WIN_LINK_FLAGS "/NODEFAULTLIB:MSVCRT")
+  ENDIF (MSVC AND NOT BUILD_SHARED_LIBS)
+ENDMACRO (TARGET_STATIC_CRT_FLAGS)
+
+#-----------------------------------------------------------------------------
+OPTION (BUILD_STATIC_CRT_LIBS "Build With Static CRT Libraries" OFF)
+IF (BUILD_STATIC_CRT_LIBS)
+  TARGET_STATIC_CRT_FLAGS ()
+ENDIF (BUILD_STATIC_CRT_LIBS)
+ 
\ No newline at end of file
Index: MANIFEST
===================================================================
--- MANIFEST	(revision 23770)
+++ MANIFEST	(revision 23771)
@@ -1361,6 +1361,8 @@
 ./tools/testfiles/topaque.h5
 ./tools/testfiles/trawdatafile.ddl
 ./tools/testfiles/trawdatafile.exp
+./tools/testfiles/trawssetfile.ddl
+./tools/testfiles/trawssetfile.exp
 ./tools/testfiles/tsaf.ddl
 ./tools/testfiles/tsaf.h5
 ./tools/testfiles/tscalarattrintsize.ddl
@@ -2243,8 +2245,12 @@
 ./config/cmake/NSIS.template.in
 ./config/cmake/NSIS.InstallOptions.ini.in
 
+# CMake-specific User Files
+./config/cmake/UserMacros/Windows_MT.cmake
+
 ./CMakeLists.txt
 ./CTestConfig.cmake
+./UserMacros.cmake
 ./c++/CMakeLists.txt
 ./c++/examples/CMakeLists.txt
 ./c++/src/CMakeLists.txt
Index: c++/test/CMakeLists.txt
===================================================================
--- c++/test/CMakeLists.txt	(revision 23770)
+++ c++/test/CMakeLists.txt	(revision 23771)
@@ -37,6 +37,7 @@
 
 ADD_EXECUTABLE (cpp_testhdf5 ${CPP_TEST_SRCS} )
 TARGET_NAMING (cpp_testhdf5 ${LIB_TYPE})
+TARGET_C_PROPERTIES (cpp_testhdf5 " " " ")
 TARGET_LINK_LIBRARIES (cpp_testhdf5
     ${HDF5_CPP_LIB_TARGET}
     ${HDF5_LIB_TARGET}
Index: c++/src/CMakeLists.txt
===================================================================
--- c++/src/CMakeLists.txt	(revision 23770)
+++ c++/src/CMakeLists.txt	(revision 23771)
@@ -82,6 +82,7 @@
 )
 
 ADD_LIBRARY (${HDF5_CPP_LIB_TARGET} ${LIB_TYPE} ${CPP_SRCS} ${CPP_HDRS})
+TARGET_C_PROPERTIES (${HDF5_CPP_LIB_TARGET} " " " ")
 TARGET_LINK_LIBRARIES (${HDF5_CPP_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_GLOBAL_VARIABLE (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_CPP_LIB_TARGET}")
 H5_SET_LIB_OPTIONS (${HDF5_CPP_LIB_TARGET} ${HDF5_CPP_LIB_NAME} ${LIB_TYPE})
Index: c++/examples/CMakeLists.txt
===================================================================
--- c++/examples/CMakeLists.txt	(revision 23770)
+++ c++/examples/CMakeLists.txt	(revision 23771)
@@ -23,6 +23,7 @@
 FOREACH (example ${examples})
   ADD_EXECUTABLE (cpp_ex_${example} ${HDF5_CPP_EXAMPLES_SOURCE_DIR}/${example}.cpp)
   TARGET_NAMING (cpp_ex_${example} ${LIB_TYPE})
+  TARGET_C_PROPERTIES (cpp_ex_${example} " " " ")
   TARGET_LINK_LIBRARIES (cpp_ex_${example} ${HDF5_CPP_LIB_TARGET} ${HDF5_LIB_TARGET})
   SET_TARGET_PROPERTIES (cpp_ex_${example} PROPERTIES FOLDER examples/cpp)
 ENDFOREACH (example ${examples})
Index: bin/reconfigure.system
===================================================================
--- bin/reconfigure.system	(revision 0)
+++ bin/reconfigure.system	(revision 23771)
@@ -0,0 +1,143 @@
+#! /bin/sh
+#
+# Copyright by the Board of Trustees of the University of Illinois.
+# All rights reserved.
+#
+# This file is part of HDF5.  The full HDF5 copyright notice, including
+# terms governing use, modification, and redistribution, is contained in
+# the files COPYING and Copyright.html.  COPYING can be found at the root
+# of the source code distribution tree; Copyright.html can be found at the
+# root level of an installed copy of the electronic HDF5 document set and
+# is linked from the top-level documents page.  It can also be found at
+# http://hdfgroup.org/HDF5/doc/Copyright.html.  If you do not have
+# access to either file, you may request a copy from help@hdfgroup.org.
+#
+
+# A script to reconfigure autotools for HDF5, and to recreate other
+# generated files specifc to HDF5.
+# If the paths of the autotools are not specified by the user, they
+# are hardcoded to point to their locations on HDF5 Linux machines.
+# Users can specify the locations of the autotools with the following
+# variables:
+# AUTOCONF, AUTOMAKE, ACLOCAL, AUTOHEADER should be the path to the
+# corresponding tools.
+# LIBTOOL_DIR should be the path to the base libtool directory;
+# $LIBTOOL_DIR/bin/libtool should invoke libtool, while
+# $LIBTOOL_DIR/share/aclocal needs to be included by aclocal.
+# Be very careful when specifying these tools manually!  There are a lot
+# of versions that can get confused (not even counting the m4 utility)!
+
+# HDF5 currently uses the following versions of the autotools:
+AUTOCONF_VERSION="autoconf (GNU Autoconf) 2.69"
+AUTOMAKE_VERSION="automake (GNU automake) 1.12.3"
+AUTOHEADER_VERSION="autoheader (GNU Autoconf) 2.69"
+ACLOCAL_VERSION="aclocal (GNU automake) 1.12.3"
+LIBTOOL_VERSION="(GNU libtool) 2.4.2"
+M4_VERSION="m4 (GNU M4) 1.4.16"
+
+#
+# When upgrading automake's version, don't forget to also update its
+# helper utilities, especially depcomp.
+
+
+# If paths to autotools are not specified by the user, assume tools are
+# running on jam in /mnt/hdf/packages and set paths accordingly.
+if test -z ${AUTOCONF}; then
+  AUTOCONF=/mnt/hdf/packages/autoconf/autoconf-2.69/bin/autoconf
+fi
+if test -z ${AUTOMAKE}; then
+  AUTOMAKE=/mnt/hdf/packages/automake/automake-1.12.3/bin/automake-1.12
+fi
+if test -z ${AUTOHEADER}; then
+  AUTOHEADER=/mnt/hdf/packages/autoconf/autoconf-2.69/bin/autoheader
+fi
+if test -z ${ACLOCAL}; then
+  ACLOCAL=/mnt/hdf/packages/automake/automake-1.12.3/bin/aclocal-1.12
+fi
+if test -z ${LIBTOOL}; then
+  LIBTOOL=/mnt/hdf/packages/libtool/libtool-2.4.2/bin/libtool
+fi
+if test -z ${M4}; then
+  M4=/mnt/hdf/packages/m4/m4-1.4.16/bin/m4
+fi
+
+# Check version numbers of all autotools against the "correct" versions
+AC_VERS=`${AUTOCONF} --version 2>&1 | grep "^${AUTOCONF_VERSION}"`
+if test -z "${AC_VERS}"; then
+   echo "${AUTOCONF} version is not ${AUTOCONF_VERSION}"
+   exit 1
+fi
+AM_VERS=`${AUTOMAKE} --version 2>&1 | grep "^${AUTOMAKE_VERSION}"`
+if test -z "${AM_VERS}"; then
+   echo "${AUTOMAKE} version is not ${AUTOMAKE_VERSION}"
+   exit 1
+fi
+AH_VERS=`${AUTOHEADER} --version 2>&1 | grep "^${AUTOHEADER_VERSION}"`
+if test -z "${AH_VERS}"; then
+   echo "${AUTOHEADER} version is not ${AUTOHEADER_VERSION}"
+   exit 1
+fi
+AL_VERS=`${ACLOCAL} --version 2>&1 | grep "^${ACLOCAL_VERSION}"`
+if test -z "${AL_VERS}"; then
+   echo "${ACLOCAL} version is not ${ACLOCAL_VERSION}"
+   exit 1
+fi
+LT_VERS=`${LIBTOOL} --version 2>&1 | grep "${LIBTOOL_VERSION}"`
+if test -z "${LT_VERS}"; then
+   echo "${LIBTOOL} version is not ${LIBTOOL_VERSION}"
+   exit 1
+fi
+M4_VERS=`${M4} --version 2>&1 | grep "${M4_VERSION}"`
+if test -z "${M4_VERS}"; then
+   echo "${M4} version is not ${M4_VERSION}"
+   exit 1
+fi
+
+# Make sure that the tools are in the path.
+AUTOCONF_DIR=`dirname ${AUTOCONF}`
+LIBTOOL_DIR=`dirname ${LIBTOOL}`
+M4_DIR=`dirname ${M4}`
+PATH=${AUTOCONF_DIR}:${M4_DIR}:$PATH
+
+# Run autoconf/automake commands in order
+  echo ${ACLOCAL} -I ${LIBTOOL_DIR}/../share/aclocal
+  ${ACLOCAL} -I ${LIBTOOL_DIR}/../share/aclocal || exit 1
+
+  echo ${AUTOHEADER}
+  ${AUTOHEADER} || exit 1
+
+  echo ${AUTOMAKE} --add-missing
+  ${AUTOMAKE} --add-missing || exit 1
+
+  echo ${AUTOCONF}
+  ${AUTOCONF} || exit 1
+
+# Clean up top-level Makefile.in
+# pmake wants an argument to be the first non-comment line it encounters
+# in the Makefile.  Automake wants to reorganize the Makefile.
+# To work around this, we post-process the top-level Makefile.in.
+  sed "s/^#xxx//" Makefile.in > Makefile.in.new
+  mv Makefile.in.new Makefile.in
+
+# Run trace script
+# The trace script adds H5TRACE macros to library source files.  It should
+# have no effect on files that don't have HDF5 API macros in them.
+echo
+echo "    Running trace script:"
+bin/trace src/H5*.c || exit 1
+
+# Run make_err
+# make_err automatically generates the H5E headers that create error message
+# types for HDF5.
+echo
+echo "    Running error generation script:"
+bin/make_err src/H5err.txt || exit 1
+
+# Run make_vers
+# make_vers automatically generates the public headers that define the API version
+# macros for HDF5.
+echo
+echo "    Running API version generation script:"
+bin/make_vers src/H5vers.txt || exit 1
+
+exit 0
Index: bin/reconfigure.local
===================================================================
--- bin/reconfigure.local	(revision 0)
+++ bin/reconfigure.local	(revision 23771)
@@ -0,0 +1,143 @@
+#! /bin/sh
+#
+# Copyright by the Board of Trustees of the University of Illinois.
+# All rights reserved.
+#
+# This file is part of HDF5.  The full HDF5 copyright notice, including
+# terms governing use, modification, and redistribution, is contained in
+# the files COPYING and Copyright.html.  COPYING can be found at the root
+# of the source code distribution tree; Copyright.html can be found at the
+# root level of an installed copy of the electronic HDF5 document set and
+# is linked from the top-level documents page.  It can also be found at
+# http://hdfgroup.org/HDF5/doc/Copyright.html.  If you do not have
+# access to either file, you may request a copy from help@hdfgroup.org.
+#
+
+# A script to reconfigure autotools for HDF5, and to recreate other
+# generated files specifc to HDF5.
+# If the paths of the autotools are not specified by the user, they
+# are hardcoded to point to their locations on HDF5 Linux machines.
+# Users can specify the locations of the autotools with the following
+# variables:
+# AUTOCONF, AUTOMAKE, ACLOCAL, AUTOHEADER should be the path to the
+# corresponding tools.
+# LIBTOOL_DIR should be the path to the base libtool directory;
+# $LIBTOOL_DIR/bin/libtool should invoke libtool, while
+# $LIBTOOL_DIR/share/aclocal needs to be included by aclocal.
+# Be very careful when specifying these tools manually!  There are a lot
+# of versions that can get confused (not even counting the m4 utility)!
+
+# HDF5 currently uses the following versions of the autotools:
+AUTOCONF_VERSION="autoconf (GNU Autoconf) 2.69"
+AUTOMAKE_VERSION="automake (GNU automake) 1.12.2"
+AUTOHEADER_VERSION="autoheader (GNU Autoconf) 2.69"
+ACLOCAL_VERSION="aclocal (GNU automake) 1.12.2"
+LIBTOOL_VERSION="(GNU libtool) 2.4.2"
+M4_VERSION="m4 (GNU M4) 1.4.16"
+
+#
+# When upgrading automake's version, don't forget to also update its
+# helper utilities, especially depcomp.
+
+
+# If paths to autotools are not specified by the user, assume tools are
+# running on jam in /mnt/hdf/packages and set paths accordingly.
+if test -z ${AUTOCONF}; then
+  AUTOCONF=/usr/bin/autoconf
+fi
+if test -z ${AUTOMAKE}; then
+  AUTOMAKE=/usr/bin/automake-1.12
+fi
+if test -z ${AUTOHEADER}; then
+  AUTOHEADER=/usr/bin/autoheader
+fi
+if test -z ${ACLOCAL}; then
+  ACLOCAL=/usr/bin/aclocal-1.12
+fi
+if test -z ${LIBTOOL}; then
+  LIBTOOL=/usr/bin/libtool
+fi
+if test -z ${M4}; then
+  M4=/usr/bin/m4
+fi
+
+# Check version numbers of all autotools against the "correct" versions
+AC_VERS=`${AUTOCONF} --version 2>&1 | grep "^${AUTOCONF_VERSION}"`
+if test -z "${AC_VERS}"; then
+   echo "${AUTOCONF} version is not ${AUTOCONF_VERSION}"
+   exit 1
+fi
+AM_VERS=`${AUTOMAKE} --version 2>&1 | grep "^${AUTOMAKE_VERSION}"`
+if test -z "${AM_VERS}"; then
+   echo "${AUTOMAKE} version is not ${AUTOMAKE_VERSION}"
+   exit 1
+fi
+AH_VERS=`${AUTOHEADER} --version 2>&1 | grep "^${AUTOHEADER_VERSION}"`
+if test -z "${AH_VERS}"; then
+   echo "${AUTOHEADER} version is not ${AUTOHEADER_VERSION}"
+   exit 1
+fi
+AL_VERS=`${ACLOCAL} --version 2>&1 | grep "^${ACLOCAL_VERSION}"`
+if test -z "${AL_VERS}"; then
+   echo "${ACLOCAL} version is not ${ACLOCAL_VERSION}"
+   exit 1
+fi
+LT_VERS=`${LIBTOOL} --version 2>&1 | grep "${LIBTOOL_VERSION}"`
+if test -z "${LT_VERS}"; then
+   echo "${LIBTOOL} version is not ${LIBTOOL_VERSION}"
+   exit 1
+fi
+M4_VERS=`${M4} --version 2>&1 | grep "${M4_VERSION}"`
+if test -z "${M4_VERS}"; then
+   echo "${M4} version is not ${M4_VERSION}"
+   exit 1
+fi
+
+# Make sure that the tools are in the path.
+AUTOCONF_DIR=`dirname ${AUTOCONF}`
+LIBTOOL_DIR=`dirname ${LIBTOOL}`
+M4_DIR=`dirname ${M4}`
+PATH=${AUTOCONF_DIR}:${M4_DIR}:$PATH
+
+# Run autoconf/automake commands in order
+  echo ${ACLOCAL} -I ${LIBTOOL_DIR}/../share/aclocal
+  ${ACLOCAL} -I ${LIBTOOL_DIR}/../share/aclocal || exit 1
+
+  echo ${AUTOHEADER}
+  ${AUTOHEADER} || exit 1
+
+  echo ${AUTOMAKE} --add-missing
+  ${AUTOMAKE} --add-missing || exit 1
+
+  echo ${AUTOCONF}
+  ${AUTOCONF} || exit 1
+
+# Clean up top-level Makefile.in
+# pmake wants an argument to be the first non-comment line it encounters
+# in the Makefile.  Automake wants to reorganize the Makefile.
+# To work around this, we post-process the top-level Makefile.in.
+  sed "s/^#xxx//" Makefile.in > Makefile.in.new
+  mv Makefile.in.new Makefile.in
+
+# Run trace script
+# The trace script adds H5TRACE macros to library source files.  It should
+# have no effect on files that don't have HDF5 API macros in them.
+echo
+echo "    Running trace script:"
+bin/trace src/H5*.c || exit 1
+
+# Run make_err
+# make_err automatically generates the H5E headers that create error message
+# types for HDF5.
+echo
+echo "    Running error generation script:"
+bin/make_err src/H5err.txt || exit 1
+
+# Run make_vers
+# make_vers automatically generates the public headers that define the API version
+# macros for HDF5.
+echo
+echo "    Running API version generation script:"
+bin/make_vers src/H5vers.txt || exit 1
+
+exit 0
Index: perform/CMakeLists.txt
===================================================================
--- perform/CMakeLists.txt	(revision 23770)
+++ perform/CMakeLists.txt	(revision 23771)
@@ -23,6 +23,7 @@
 )
 ADD_EXECUTABLE (h5perf_serial ${h5perf_serial_SRCS})
 TARGET_NAMING (h5perf_serial ${LIB_TYPE})
+TARGET_C_PROPERTIES (h5perf_serial " " " ")
 TARGET_LINK_LIBRARIES (h5perf_serial ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
 SET_TARGET_PROPERTIES (h5perf_serial PROPERTIES FOLDER perform)
 
@@ -38,6 +39,7 @@
       APPEND PROPERTY COMPILE_DEFINITIONS STANDALONE
   )
   TARGET_NAMING (h5perf_serial_alone ${LIB_TYPE})
+  TARGET_C_PROPERTIES (h5perf_serial_alone " " " ")
   TARGET_LINK_LIBRARIES (h5perf_serial_alone ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
   SET_TARGET_PROPERTIES (h5perf_serial_alone PROPERTIES FOLDER perform)
 ENDIF (HDF5_BUILD_PERFORM_STANDALONE)
@@ -48,6 +50,7 @@
 )
 ADD_EXECUTABLE(chunk ${chunk_SRCS})
 TARGET_NAMING (chunk ${LIB_TYPE})
+TARGET_C_PROPERTIES (chunk " " " ")
 TARGET_LINK_LIBRARIES(chunk ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
 SET_TARGET_PROPERTIES (chunk PROPERTIES FOLDER perform)
 
@@ -57,6 +60,7 @@
 )
 ADD_EXECUTABLE (iopipe ${iopipe_SRCS})
 TARGET_NAMING (iopipe ${LIB_TYPE})
+TARGET_C_PROPERTIES (iopipe " " " ")
 TARGET_LINK_LIBRARIES (iopipe ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
 SET_TARGET_PROPERTIES (iopipe PROPERTIES FOLDER perform)
 
@@ -66,6 +70,7 @@
 )
 ADD_EXECUTABLE (overhead ${overhead_SRCS})
 TARGET_NAMING (overhead ${LIB_TYPE})
+TARGET_C_PROPERTIES (overhead " " " ")
 TARGET_LINK_LIBRARIES (overhead ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET})
 SET_TARGET_PROPERTIES (overhead PROPERTIES FOLDER perform)
 
@@ -75,6 +80,7 @@
 )
 ADD_EXECUTABLE (perf_meta ${perf_meta_SRCS})
 TARGET_NAMING (perf_meta ${LIB_TYPE})
+TARGET_C_PROPERTIES (perf_meta " " " ")
 TARGET_LINK_LIBRARIES (perf_meta ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
 SET_TARGET_PROPERTIES (perf_meta PROPERTIES FOLDER perform)
 
@@ -84,6 +90,7 @@
 )
 ADD_EXECUTABLE (zip_perf ${zip_perf_SRCS})
 TARGET_NAMING (zip_perf ${LIB_TYPE})
+TARGET_C_PROPERTIES (zip_perf " " " ")
 TARGET_LINK_LIBRARIES (zip_perf ${HDF5_TOOLS_LIB_TARGET} ${HDF5_LIB_TARGET})
 SET_TARGET_PROPERTIES (zip_perf PROPERTIES FOLDER perform)
 
@@ -96,6 +103,7 @@
   )
   ADD_EXECUTABLE (h5perf ${h5perf_SRCS})
   TARGET_NAMING (h5perf ${LIB_TYPE})
+  TARGET_C_PROPERTIES (h5perf " " " ")
   TARGET_LINK_LIBRARIES (h5perf ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
   SET_TARGET_PROPERTIES (h5perf PROPERTIES FOLDER perform)
 
@@ -111,6 +119,7 @@
         APPEND PROPERTY COMPILE_DEFINITIONS STANDALONE
     )
     TARGET_NAMING (h5perf_alone ${LIB_TYPE})
+    TARGET_C_PROPERTIES (h5perf_alone " " " ")
     TARGET_LINK_LIBRARIES (h5perf_alone ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
     SET_TARGET_PROPERTIES (h5perf_alone PROPERTIES FOLDER perform)
   ENDIF (HDF5_BUILD_PERFORM_STANDALONE)
@@ -122,6 +131,7 @@
     )
     ADD_EXECUTABLE (benchpar ${benchpar_SRCS})
     TARGET_NAMING (benchpar ${LIB_TYPE})
+    TARGET_C_PROPERTIES (benchpar " " " ")
     TARGET_LINK_LIBRARIES (benchpar ${HDF5_LIB_TARGET} ${HDF5_TOOLS_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
     SET_TARGET_PROPERTIES (benchpar PROPERTIES FOLDER perform)
   ENDIF (HDF5_BUILD_PARALLEL_ALL)
Index: fortran/test/CMakeLists.txt
===================================================================
--- fortran/test/CMakeLists.txt	(revision 23770)
+++ fortran/test/CMakeLists.txt	(revision 23771)
@@ -10,6 +10,7 @@
 # Add Test Lib
 #-----------------------------------------------------------------------------
 ADD_LIBRARY (${HDF5_F90_C_TEST_LIB_TARGET} ${LIB_TYPE} t.c)
+TARGET_C_PROPERTIES (${HDF5_F90_C_TEST_LIB_TARGET} " " " ")
 TARGET_LINK_LIBRARIES (${HDF5_F90_C_TEST_LIB_TARGET}
     ${HDF5_F90_C_LIB_TARGET}
     ${HDF5_TEST_LIB_TARGET}
@@ -28,7 +29,7 @@
   ENDIF (BUILD_SHARED_LIBS)
   SET_PROPERTY (TARGET ${HDF5_F90_TEST_LIB_TARGET} APPEND PROPERTY COMPILE_DEFINITIONS HDF5F90_WINDOWS)
 ENDIF (WIN32 AND NOT CYGWIN)
-TARGET_FORTRAN_WIN_PROPERTIES (${HDF5_F90_TEST_LIB_TARGET} ${SHARED_LINK_FLAGS})
+TARGET_FORTRAN_PROPERTIES (${HDF5_F90_TEST_LIB_TARGET} " " ${SHARED_LINK_FLAGS})
 SET_TARGET_PROPERTIES (${HDF5_F90_TEST_LIB_TARGET} PROPERTIES LINKER_LANGUAGE Fortran)
 TARGET_LINK_LIBRARIES (${HDF5_F90_TEST_LIB_TARGET}
     ${HDF5_F90_C_TEST_LIB_TARGET}
@@ -60,7 +61,7 @@
     tH5Z.f90
 )
 TARGET_NAMING (testhdf5_fortran ${LIB_TYPE})
-TARGET_FORTRAN_WIN_PROPERTIES (testhdf5_fortran "")
+TARGET_FORTRAN_PROPERTIES (testhdf5_fortran " " " ")
 TARGET_LINK_LIBRARIES (testhdf5_fortran 
     ${HDF5_F90_TEST_LIB_TARGET}
     ${HDF5_F90_LIB_TARGET}
@@ -84,7 +85,7 @@
     tH5G_1_8.f90
 )
 TARGET_NAMING (testhdf5_fortran_1_8 ${LIB_TYPE})
-TARGET_FORTRAN_WIN_PROPERTIES (testhdf5_fortran_1_8 "")
+TARGET_FORTRAN_PROPERTIES (testhdf5_fortran_1_8 " " " ")
 TARGET_LINK_LIBRARIES (testhdf5_fortran_1_8 
     ${HDF5_F90_TEST_LIB_TARGET}
     ${HDF5_F90_LIB_TARGET}
@@ -112,7 +113,7 @@
       tH5T_F03.f90
   )
   TARGET_NAMING (fortranlib_test_F03 ${LIB_TYPE})
-  TARGET_FORTRAN_WIN_PROPERTIES (fortranlib_test_F03 "")
+  TARGET_FORTRAN_PROPERTIES (fortranlib_test_F03 " " " ")
   TARGET_LINK_LIBRARIES (fortranlib_test_F03 
       ${HDF5_F90_TEST_LIB_TARGET}
       ${HDF5_F90_LIB_TARGET}
@@ -131,7 +132,7 @@
 #-- Adding test for fflush1
 ADD_EXECUTABLE (fflush1 fflush1.f90)
 TARGET_NAMING (fflush1 ${LIB_TYPE})
-TARGET_FORTRAN_WIN_PROPERTIES (fflush1 "")
+TARGET_FORTRAN_PROPERTIES (fflush1 " " " ")
 TARGET_LINK_LIBRARIES (fflush1 
     ${HDF5_F90_LIB_TARGET}
     ${HDF5_F90_TEST_LIB_TARGET}
@@ -148,7 +149,7 @@
 #-- Adding test for fflush2
 ADD_EXECUTABLE (fflush2 fflush2.f90)
 TARGET_NAMING (fflush2 ${LIB_TYPE})
-TARGET_FORTRAN_WIN_PROPERTIES (fflush2 "")
+TARGET_FORTRAN_PROPERTIES (fflush2 " " " ")
 TARGET_LINK_LIBRARIES (fflush2 
     ${HDF5_F90_TEST_LIB_TARGET}
     ${HDF5_F90_LIB_TARGET}
Index: fortran/testpar/CMakeLists.txt
===================================================================
--- fortran/testpar/CMakeLists.txt	(revision 23770)
+++ fortran/testpar/CMakeLists.txt	(revision 23771)
@@ -17,7 +17,7 @@
     mdset.f90
 )
 TARGET_NAMING (parallel_test ${LIB_TYPE})
-TARGET_FORTRAN_WIN_PROPERTIES (parallel_test "")
+TARGET_FORTRAN_PROPERTIES (parallel_test " " " ")
 TARGET_LINK_LIBRARIES (parallel_test 
     ${HDF5_F90_TEST_LIB_TARGET}
     ${HDF5_F90_LIB_TARGET}
Index: fortran/src/CMakeLists.txt
===================================================================
--- fortran/src/CMakeLists.txt	(revision 23770)
+++ fortran/src/CMakeLists.txt	(revision 23771)
@@ -133,6 +133,7 @@
 )
 
 ADD_LIBRARY (${HDF5_F90_C_LIB_TARGET} ${LIB_TYPE} ${f90CStub_C_SRCS} ${f90CStub_C_HDRS})
+TARGET_C_PROPERTIES (${HDF5_F90_C_LIB_TARGET} " " " ")
 TARGET_LINK_LIBRARIES (${HDF5_F90_C_LIB_TARGET} ${HDF5_LIB_TARGET} ${LINK_LIBS})
 SET_GLOBAL_VARIABLE (HDF5_LIBRARIES_TO_EXPORT "${HDF5_LIBRARIES_TO_EXPORT};${HDF5_F90_C_LIB_TARGET}")
 H5_SET_LIB_OPTIONS (${HDF5_F90_C_LIB_TARGET} ${HDF5_F90_C_LIB_NAME} ${LIB_TYPE})
@@ -233,7 +234,7 @@
         HDF5F90_WINDOWS
   )
 ENDIF (WIN32 AND NOT CYGWIN)
-TARGET_FORTRAN_WIN_PROPERTIES (${HDF5_F90_LIB_TARGET} ${SHARED_LINK_FLAGS})
+TARGET_FORTRAN_PROPERTIES (${HDF5_F90_LIB_TARGET} " " ${SHARED_LINK_FLAGS})
 SET_TARGET_PROPERTIES (${HDF5_F90_LIB_TARGET} PROPERTIES LINKER_LANGUAGE Fortran)
 TARGET_LINK_LIBRARIES (${HDF5_F90_LIB_TARGET} ${HDF5_F90_C_LIB_TARGET} ${HDF5_LIB_TARGET})
 IF (H5_HAVE_PARALLEL AND MPI_Fortran_FOUND)
Index: fortran/examples/CMakeLists.txt
===================================================================
--- fortran/examples/CMakeLists.txt	(revision 23770)
+++ fortran/examples/CMakeLists.txt	(revision 23771)
@@ -41,7 +41,7 @@
 FOREACH (example ${examples})
   ADD_EXECUTABLE (f90_ex_${example} ${HDF5_F90_EXAMPLES_SOURCE_DIR}/${example}.f90)
   TARGET_NAMING (f90_ex_${example} ${LIB_TYPE})
-  TARGET_FORTRAN_WIN_PROPERTIES (f90_ex_${example} "")
+  TARGET_FORTRAN_PROPERTIES (f90_ex_${example} " " " ")
   IF (WIN32 AND NOT CYGWIN)
     SET_PROPERTY (TARGET f90_ex_${example} 
         APPEND PROPERTY COMPILE_DEFINITIONS 
@@ -69,7 +69,7 @@
   FOREACH (example ${F2003_examples})
     ADD_EXECUTABLE (f03_ex_${example} ${HDF5_F90_EXAMPLES_SOURCE_DIR}/${example}.f90)
     TARGET_NAMING (f03_ex_${example} ${LIB_TYPE})
-    TARGET_FORTRAN_WIN_PROPERTIES (f03_ex_${example} "")
+    TARGET_FORTRAN_PROPERTIES (f03_ex_${example} " " " ")
     IF (WIN32 AND NOT CYGWIN)
       SET_PROPERTY (TARGET f03_ex_${example} 
           APPEND PROPERTY COMPILE_DEFINITIONS HDF5F90_WINDOWS
@@ -95,7 +95,7 @@
 IF (H5_HAVE_PARALLEL AND MPI_Fortran_FOUND)
   ADD_EXECUTABLE (f90_ex_ph5example ${HDF5_F90_EXAMPLES_SOURCE_DIR}/ph5example.f90)
   TARGET_NAMING (f90_ex_ph5example ${LIB_TYPE})
-  TARGET_FORTRAN_WIN_PROPERTIES (f90_ex_ph5example "")
+  TARGET_FORTRAN_PROPERTIES (f90_ex_ph5example " " " ")
   IF (WIN32 AND NOT CYGWIN)
     SET_PROPERTY (TARGET f90_ex_ph5example 
         APPEND PROPERTY COMPILE_DEFINITIONS 
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 23770)
+++ CMakeLists.txt	(revision 23771)
@@ -176,7 +176,7 @@
     "\\1" H5_VERS_MINOR ${_h5public_h_contents})
 STRING (REGEX REPLACE ".*#define[ \t]+H5_VERS_RELEASE[ \t]+([0-9]*).*$"
     "\\1" H5_VERS_RELEASE ${_h5public_h_contents})
-STRING (REGEX REPLACE ".*#define[ \t]+H5_VERS_SUBRELEASE[ \t]+\"([0-9A-Za-z.]*)\".*$"
+STRING (REGEX REPLACE ".*#define[ \t]+H5_VERS_SUBRELEASE[ \t]+\"([0-9A-Za-z._]*)\".*$"
     "\\1" H5_VERS_SUBRELEASE ${_h5public_h_contents})
 #MESSAGE (STATUS "VERSION: ${H5_VERS_MAJOR}.${H5_VERS_MINOR}.${H5_VERS_RELEASE}-${H5_VERS_SUBRELEASE}")
 
@@ -190,7 +190,7 @@
     "\\1" H5_SOVERS_MINOR ${_lt_vers_am_contents})
 STRING (REGEX REPLACE ".*LT_VERS_AGE[ \t]+=[ \t]+([0-9]*).*$"
     "\\1" H5_SOVERS_RELEASE ${_lt_vers_am_contents})
-MESSAGE (STATUS "SOVERSION: ${H5_SOVERS_MAJOR}.${H5_SOVERS_MINOR}.${H5_SOVERS_RELEASE}")
+MESSAGE (STATUS "SOVERSION: ${H5_SOVERS_MAJOR}.${H5_SOVERS_RELEASE}.${H5_SOVERS_MINOR}")
 
 #-----------------------------------------------------------------------------
 # Basic HDF5 stuff here
@@ -205,7 +205,7 @@
 ELSE (NOT "${H5_VERS_SUBRELEASE}" STREQUAL "")
   SET (HDF5_PACKAGE_VERSION_STRING "${HDF5_PACKAGE_VERSION}")
 ENDIF (NOT "${H5_VERS_SUBRELEASE}" STREQUAL "")
-SET (HDF5_PACKAGE_SOVERSION "${H5_SOVERS_MAJOR}.${H5_SOVERS_MINOR}.${H5_SOVERS_RELEASE}")
+SET (HDF5_PACKAGE_SOVERSION "${H5_SOVERS_MAJOR}.${H5_SOVERS_RELEASE}.${H5_SOVERS_MINOR}")
 SET (HDF5_PACKAGE_STRING "${HDF5_PACKAGE_NAME} ${HDF5_PACKAGE_VERSION_STRING}")
 SET (HDF5_PACKAGE_TARNAME "${HDF5_PACKAGE}${HDF_PACKAGE_EXT}")
 SET (HDF5_PACKAGE_URL "http://www.hdfgroup.org")
@@ -358,6 +358,8 @@
 
 IF (MSVC)
   SET (CMAKE_MFC_FLAG 0)
+  SET (WIN_COMPILE_FLAGS "/MD")
+  SET (WIN_LINK_FLAGS "")
 ENDIF (MSVC)
 
 SET (MAKE_SYSTEM)
@@ -519,6 +521,11 @@
 ENDIF (HDF5_USE_16_API_DEFAULT)
 
 #-----------------------------------------------------------------------------
+# Include user macros
+#-----------------------------------------------------------------------------
+INCLUDE (UserMacros.cmake)
+
+#-----------------------------------------------------------------------------
 # Options for HDF5 Filters
 #-----------------------------------------------------------------------------
 MACRO (HDF5_SETUP_FILTERS FILTER)
Index: examples/CMakeLists.txt
===================================================================
--- examples/CMakeLists.txt	(revision 23770)
+++ examples/CMakeLists.txt	(revision 23771)
@@ -39,6 +39,7 @@
 FOREACH (example ${examples})
   ADD_EXECUTABLE (${example} ${HDF5_EXAMPLES_SOURCE_DIR}/${example}.c)
   TARGET_NAMING (${example} ${LIB_TYPE})
+  TARGET_C_PROPERTIES (${example} " " " ")
   TARGET_LINK_LIBRARIES (${example} ${HDF5_LIB_TARGET})
   SET_TARGET_PROPERTIES (${example} PROPERTIES FOLDER examples)
 ENDFOREACH (example ${examples})
@@ -46,6 +47,7 @@
 IF (H5_HAVE_PARALLEL)
   ADD_EXECUTABLE (ph5example ${HDF5_EXAMPLES_SOURCE_DIR}/ph5example.c)
   TARGET_NAMING (ph5example ${LIB_TYPE})
+  TARGET_C_PROPERTIES (ph5example " " " ")
   TARGET_LINK_LIBRARIES (ph5example ${HDF5_LIB_TARGET})
   SET_TARGET_PROPERTIES (ph5example PROPERTIES FOLDER examples)
 ENDIF (H5_HAVE_PARALLEL)
