diff -ruN hdf5-1.8.11.orig/compile_windows_i386.bat hdf5-1.8.11/compile_windows_i386.bat
--- compile_windows_i386.bat	1970-01-01 01:00:00.000000000 +0100
+++ compile_windows_i386.bat	2013-06-22 00:04:09.000000000 +0200
@@ -0,0 +1,6 @@
+@echo off
+rd /S /Q build
+mkdir build
+cd build
+cmake -G "Visual Studio 9 2008" -DBUILD_SHARED_LIBS:BOOL=OFF -DHDF5_ENABLE_Z_LIB_SUPPORT:BOOL=ON -DHDF5_ALLOW_EXTERNAL_SUPPORT:STRING="TGZ" -DZLIB_TGZ_NAME:STRING="zlib-1.2.8.tar.gz" -DTGZ_PATH:STRING="u:\winbuild" ..
+cmake --build . --config Release
diff -ruN hdf5-1.8.11.orig/compile_windows_x64.bat hdf5-1.8.11/compile_windows_x64.bat
--- compile_windows_x64.bat	1970-01-01 01:00:00.000000000 +0100
+++ compile_windows_x64.bat	2013-06-22 00:05:05.000000000 +0200
@@ -0,0 +1,6 @@
+@echo off
+rd /S /Q build
+mkdir build
+cd build
+cmake -G "Visual Studio 9 2008 Win64" -DBUILD_SHARED_LIBS:BOOL=OFF -DHDF5_ENABLE_Z_LIB_SUPPORT:BOOL=ON -DHDF5_ALLOW_EXTERNAL_SUPPORT:STRING="TGZ" -DZLIB_TGZ_NAME:STRING="zlib-1.2.8.tar.gz" -DTGZ_PATH:STRING="u:\winbuild" ..
+cmake --build . --config Release
