--- hdf5-1.10.3-pre1/java/src/jni/exceptionImp.c	2018-07-29 14:25:43.000000000 +0200
+++ hdf5-1.10.3-pre1/java/src/jni/exceptionImp.c	2018-08-03 23:03:03.973908021 +0200
@@ -86,6 +86,10 @@
         return JNI_FALSE;                                                   \
     }                                                                       \
     ex = ENVPTR->NewObjectA (ENVPAR jc, jm, (jvalue*)(args));               \
+    if (ex == NULL) {                                                       \
+        printf("FATAL ERROR:  %s: Creation failed\n", (className));         \
+        return JNI_FALSE;                                                   \
+    }                                                                       \
     if (ENVPTR->Throw(ENVPAR (jthrowable)ex) < 0) {                         \
         printf("FATAL ERROR:  %s: Throw failed\n", (className));            \
         return JNI_FALSE;                                                   \
@@ -174,13 +178,13 @@
 
 /*
  * Class:     hdf_hdf5lib_exceptions_HDFLibraryException
- * Method:    getMajorErrorNumber
+ * Method:    _getMajorErrorNumber
  * Signature: ()J
  *
  *  Extract the HDF-5 major error number from the HDF-5 error stack.
  */
 JNIEXPORT jlong JNICALL
-Java_hdf_hdf5lib_exceptions_HDF5LibraryException_getMajorErrorNumber
+Java_hdf_hdf5lib_exceptions_HDF5LibraryException__1getMajorErrorNumber
     (JNIEnv *env, jobject obj)
 {
     H5E_num_t err_nums;
@@ -190,17 +194,17 @@
     H5Ewalk2(H5E_DEFAULT, H5E_WALK_DOWNWARD, walk_error_callback, &err_nums);
 
     return err_nums.maj_num;
-} /* end Java_hdf_hdf5lib_exceptions_HDF5LibraryException_getMajorErrorNumber() */
+} /* end Java_hdf_hdf5lib_exceptions_HDF5LibraryException__1getMajorErrorNumber() */
 
 /*
  * Class:     hdf_hdf5lib_exceptions_HDFLibraryException
- * Method:    getMinorErrorNumber
+ * Method:    _getMinorErrorNumber
  * Signature: ()J
  *
  *  Extract the HDF-5 minor error number from the HDF-5 error stack.
  */
 JNIEXPORT jlong JNICALL
-Java_hdf_hdf5lib_exceptions_HDF5LibraryException_getMinorErrorNumber
+Java_hdf_hdf5lib_exceptions_HDF5LibraryException__1getMinorErrorNumber
     (JNIEnv *env, jobject obj)
 {
     H5E_num_t err_nums;
@@ -210,7 +214,7 @@
     H5Ewalk2(H5E_DEFAULT, H5E_WALK_DOWNWARD, walk_error_callback, &err_nums);
 
     return err_nums.min_num;
-} /* end Java_hdf_hdf5lib_exceptions_HDF5LibraryException_getMinorErrorNumber() */
+} /* end Java_hdf_hdf5lib_exceptions_HDF5LibraryException__1getMinorErrorNumber() */
 
 /*
  *  Routine to raise particular Java exceptions from C
