diff -ruN hdf5-1.8.8-pre1.orig/configure hdf5-1.8.8-pre1/configure
--- configure.orig	2011-10-27 16:16:25.000000000 +0200
+++ configure	2011-10-30 10:27:50.000000000 +0100
@@ -22692,7 +22692,6 @@
                                                                 H5_CPPFLAGS="-D_POSIX_C_SOURCE=199506L $H5_CPPFLAGS"
 
                             AM_CPPFLAGS="-D_BSD_SOURCE $AM_CPPFLAGS"
-    AM_LDFLAGS="-lbsd-compat $AM_LDFLAGS"
     ;;
 esac
 
diff -ruN hdf5-1.8.8-pre1.orig/configure.in hdf5-1.8.8-pre1/configure.in
--- configure.in.orig	2011-10-27 16:16:25.000000000 +0200
+++ configure.in	2011-10-30 10:30:15.000000000 +0100
@@ -1495,7 +1495,6 @@
     dnl Linking to the bsd-compat library is required as per the gcc manual:
     dnl http://www.gnu.org/s/libc/manual/html_node/Feature-Test-Macros.html
     AM_CPPFLAGS="-D_BSD_SOURCE $AM_CPPFLAGS"
-    AM_LDFLAGS="-lbsd-compat $AM_LDFLAGS"
     ;;
 esac
 
